In [119]:
# ! pip install pyarrow

In [2]:
import pandas as pd
import pyarrow as pa
import pyarrow.parquet as pq

In [3]:
# 데이터 원본
test_series = pq.read_table('/Users/soo/Downloads/child/test_series.parquet').to_pandas()
test_series

series_id  step                 timestamp     anglez    enmo
0    038441c925bb     0  2018-08-14T15:30:00-0400   2.636700  0.0217
1    038441c925bb     1  2018-08-14T15:30:05-0400   2.636800  0.0215
2    038441c925bb     2  2018-08-14T15:30:10-0400   2.637000  0.0216
3    038441c925bb     3  2018-08-14T15:30:15-0400   2.636800  0.0213
4    038441c925bb     4  2018-08-14T15:30:20-0400   2.636800  0.0215
..            ...   ...                       ...        ...     ...
445  0402a003dae9   145  2018-12-18T12:57:05-0500 -59.696899  0.0601
446  0402a003dae9   146  2018-12-18T12:57:10-0500 -35.656601  0.0427
447  0402a003dae9   147  2018-12-18T12:57:15-0500 -21.582399  0.0309
448  0402a003dae9   148  2018-12-18T12:57:20-0500 -42.616001  0.0328
449  0402a003dae9   149  2018-12-18T12:57:25-0500   7.029900  0.0081

[450 rows x 5 columns]

In [24]:
train_series.loc[(train_series['series_id']== '038441c925bb') & (train_series['step'] >= 4990) & (train_series['step'] <= 10935), :]

series_id   step                 timestamp     anglez    enmo
4990   038441c925bb   4990  2018-08-14T22:25:50-0400 -78.709801  0.0097
4991   038441c925bb   4991  2018-08-14T22:25:55-0400 -78.730301  0.0104
4992   038441c925bb   4992  2018-08-14T22:26:00-0400 -78.690598  0.0099
4993   038441c925bb   4993  2018-08-14T22:26:05-0400 -78.664902  0.0099
4994   038441c925bb   4994  2018-08-14T22:26:10-0400 -78.465897  0.0101
...             ...    ...                       ...        ...     ...
10931  038441c925bb  10931  2018-08-15T06:40:55-0400 -61.437500  0.0267
10932  038441c925bb  10932  2018-08-15T06:41:00-0400 -61.578201  0.0263
10933  038441c925bb  10933  2018-08-15T06:41:05-0400 -61.744301  0.0264
10934  038441c925bb  10934  2018-08-15T06:41:10-0400 -61.786400  0.0267
10935  038441c925bb  10935  2018-08-15T06:41:15-0400 -61.757198  0.0266

[5946 rows x 5 columns]

In [5]:
# 데이터 원본
train_series = pq.read_table('/Users/soo/Downloads/child/train_series.parquet').to_pandas()
train_series

series_id    step                 timestamp     anglez    enmo
0          038441c925bb       0  2018-08-14T15:30:00-0400   2.636700  0.0217
1          038441c925bb       1  2018-08-14T15:30:05-0400   2.636800  0.0215
2          038441c925bb       2  2018-08-14T15:30:10-0400   2.637000  0.0216
3          038441c925bb       3  2018-08-14T15:30:15-0400   2.636800  0.0213
4          038441c925bb       4  2018-08-14T15:30:20-0400   2.636800  0.0215
...                 ...     ...                       ...        ...     ...
127946335  fe90110788d2  592375  2017-09-08T00:14:35-0400 -27.277500  0.0204
127946336  fe90110788d2  592376  2017-09-08T00:14:40-0400 -27.032499  0.0233
127946337  fe90110788d2  592377  2017-09-08T00:14:45-0400 -26.841200  0.0202
127946338  fe90110788d2  592378  2017-09-08T00:14:50-0400 -26.723900  0.0199
127946339  fe90110788d2  592379  2017-09-08T00:14:55-0400 -31.521601  0.0205

[127946340 rows x 5 columns]

In [5]:
# train_series[train_series['step']==4992]

In [6]:
# 데이터 원본
train_events = pd.read_csv('/Users/soo/Downloads/child/train_events.csv')
train_events

series_id  night   event      step                 timestamp
0      038441c925bb      1   onset    4992.0  2018-08-14T22:26:00-0400
1      038441c925bb      1  wakeup   10932.0  2018-08-15T06:41:00-0400
2      038441c925bb      2   onset   20244.0  2018-08-15T19:37:00-0400
3      038441c925bb      2  wakeup   27492.0  2018-08-16T05:41:00-0400
4      038441c925bb      3   onset   39996.0  2018-08-16T23:03:00-0400
...             ...    ...     ...       ...                       ...
14503  fe90110788d2     33  wakeup  560604.0  2017-09-06T04:07:00-0400
14504  fe90110788d2     34   onset  574620.0  2017-09-06T23:35:00-0400
14505  fe90110788d2     34  wakeup  581604.0  2017-09-07T09:17:00-0400
14506  fe90110788d2     35   onset       NaN                       NaN
14507  fe90110788d2     35  wakeup       NaN                       NaN

[14508 rows x 5 columns]

In [7]:
#총 유저 
print('총 유저 : ',len(train_events['series_id'].unique()),'명')
print(train_events['series_id'].unique())

총 유저 :  277 명
['038441c925bb' '03d92c9f6f8a' '0402a003dae9' '04f547b8017d'
 '05e1944c3818' '062cae666e2a' '062dbd4c95e6' '08db4255286f'
 '0a96f4993bd7' '0cd1e3d0ed95' '0ce74d6d2106' '0cfc06c129cc'
 '0d0ad1e77851' '0dee4fda51c3' '0ec9fc461819' '0ef7d94fde99'
 '0f572d690310' '0f9e60a8e56d' '10469f6765bf' '1087d7b0ff2e'
 '10f8bc1f7b07' '12d01911d509' '1319a1935f48' '137771d19ca2'
 '137b99e936ab' '13b4d6a01d27' '148471991ffb' '154fe824ed87'
 '16fe2798ed0f' '1716cd4163b2' '1762ab70ec76' '188d4b7cd28b'
 '18a0ca03431d' '18b61dd5aae8' '1955d568d987' '1b92be89db4c'
 '1c7c0bad1263' '1d4569cbac0f' '1e6717d93c1d' '1f96b9668bdf'
 '207eded97727' '25e2b3dd9c3b' '2654a87be968' '27f09a6a858f'
 '280e08693c6d' '292a75c0b94e' '29c75c018220' '29d3469bd15d'
 '2b0a1fa8eba8' '2b8d87addea9' '2cd2340ca14d' '2e9ced2c7976'
 '2f7504d0f426' '2fbbee1a38e3' '2fc653ca75c7' '31011ade7c0a'
 '3318a0e3ed6f' '33ceeba8918a' '3452b878e596' '349c5562ee2c'
 '35826366dfc7' '361366da569e' '3664fe9233f9' '3665c86afaf5'
 '390b4872

In [8]:
#train_events 원본's 유저당 데이터 갯수 샘플
print('038441c925bb 데이터 갯수 : ',len(train_events[train_events['series_id'] == '038441c925bb']))
print('03d92c9f6f8a 데이터 갯수 : ',len(train_events[train_events['series_id'] == '03d92c9f6f8a']))
print('0402a003dae9 데이터 갯수 : ',len(train_events[train_events['series_id'] == '0402a003dae9']))
print('04f547b8017d 데이터 갯수 : ',len(train_events[train_events['series_id'] == '04f547b8017d']))
print('05e1944c3818 데이터 갯수 : ',len(train_events[train_events['series_id'] == '05e1944c3818']))

038441c925bb 데이터 갯수 :  46
03d92c9f6f8a 데이터 갯수 :  74
0402a003dae9 데이터 갯수 :  48
04f547b8017d 데이터 갯수 :  74
05e1944c3818 데이터 갯수 :  16


### 결측값 데이터 확인하기 1

* 총 유저 : 277 명
* 결측값 있는 유저  : 240 명
* 결측값이 전혀 없는 유저 : 37 명 <br>
['939932f1822d', '3664fe9233f9', '76237b9406d5', '31011ade7c0a', '72bbd1ac3edf', 'b364205aba43', '655f19eabf1e', '483d6545417f', '29c75c018220', 'd5e47b94477e', '08db4255286f', '18b61dd5aae8', '89bd631d1769', 'd150801f3145', 'c535634d7dcd', '7822ee8fe3ec', '3452b878e596', '349c5562ee2c', 'd25e479ecbb7', 'af91d9a50547', '55a47ff9dc8a', 'ca730dbf521d', 'a596ad0b82aa', '10f8bc1f7b07', 'c68260cc9e8f', '67f5fc60e494', 'a9e5f5314bcb', '5f94bb3e1bed', '0a96f4993bd7', 'c6788e579967', '8e32047cbc1f', 'a9a2f7fac455', '1087d7b0ff2e', '9ee455e4770d', '5acc9d63b5fd', '0cfc06c129cc', 'd515236bdeec']

* 결측값 없는 유저 (37명)의 train_events 데이터 수 : TTL 14,508 -> 1,562
<br>=> 데이터 : 1/10로 감소
* 결측값 없는 유저 (37명)의 train_series 데이터 수 : TTL 127,946,340 -> 14,334,840
<br>=> 데이터 : 1/10로 감소

In [9]:
drop_user = []
for index, row in train_events.iterrows():
    if train_events.loc[index].isnull().any():
        drop_user.append(train_events['series_id'][index])

drop_user = list(set(drop_user))
print('결측값 포함한 유저 :',len(drop_user),'명')
print(drop_user)

결측값 포함한 유저 : 240 명
['db5e0ee1c0ab', 'f0482490923c', '55b7f5c99930', '785c9ca4eff7', '702bb5387b1e', 'cfeb11428dd7', '2fc653ca75c7', '148471991ffb', '2654a87be968', 'a81f4472c637', '72ba4a8afff4', '2e9ced2c7976', '7476c0bd18d2', 'd043c0ca71cd', 'c8053490cec2', '0402a003dae9', 'e69aff66e0cb', 'fb223ed2278c', 'fe90110788d2', 'b750c8c1556c', 'f8a8da8bdd00', 'e0686434d029', '29d3469bd15d', '752900afe3a6', 'bf00506437aa', '5f40907ec171', 'd2d6b9af0553', 'fa149c3c4bde', '0f9e60a8e56d', '390b487231ce', '3aceb17ef7bd', 'e4500e7e19e1', 'e1f2a4f991cb', 'def21f50dd3c', '3c336d6ba566', '8a22387617c3', 'c7b2155a4a47', 'e6ddbaaf0639', '51b23d177971', '0f572d690310', 'd2fef7e4defd', '7504165f497d', 'f564985ab692', '18a0ca03431d', '83fa182bec3a', 'f7eb179216c2', '1716cd4163b2', '72d2234e84e4', '91127c2b0e60', 'bb5612895813', 'd5be621fd9aa', '25e2b3dd9c3b', '4a31811f3558', '4ab54be1a403', '60d31b0bec3b', 'df33ae359fb5', 'ebb6fae8ed43', '9aed9ee12ae2', '416354edd92a', 'c289c8a823e0', '4743bdde25df', 'fbf

In [10]:
noNan_user = []
for index, row in train_events.iterrows():
    if not train_events.loc[index].isin(drop_user).any():
        noNan_user.append(train_events['series_id'][index])

noNan_user = list(set(noNan_user))
# noNan_user_set = set(noNan_user)

print('결측값이 없는 유저 :', len(noNan_user), '명')
print(noNan_user)

결측값이 없는 유저 : 37 명
['31011ade7c0a', '7822ee8fe3ec', '89bd631d1769', '0cfc06c129cc', 'd25e479ecbb7', 'd150801f3145', '655f19eabf1e', '3664fe9233f9', '29c75c018220', '10f8bc1f7b07', '939932f1822d', 'a9a2f7fac455', 'ca730dbf521d', 'a9e5f5314bcb', '483d6545417f', '9ee455e4770d', '08db4255286f', '18b61dd5aae8', 'c68260cc9e8f', '349c5562ee2c', '5acc9d63b5fd', '55a47ff9dc8a', '72bbd1ac3edf', '76237b9406d5', 'd515236bdeec', '5f94bb3e1bed', '3452b878e596', '1087d7b0ff2e', 'b364205aba43', 'a596ad0b82aa', 'c6788e579967', '67f5fc60e494', '0a96f4993bd7', 'd5e47b94477e', 'af91d9a50547', '8e32047cbc1f', 'c535634d7dcd']


In [11]:
drop_train_events=train_events[train_events['series_id'].isin(noNan_user)]
# drop_train_events = train_events[train_events['series_id'].isin(noNan_user_set)]
drop_train_events

series_id  night   event      step                 timestamp
398    08db4255286f      1   onset   11988.0  2018-11-06T02:39:00-0400
399    08db4255286f      1  wakeup   14388.0  2018-11-06T05:59:00-0400
400    08db4255286f      2   onset   28680.0  2018-11-07T01:50:00-0400
401    08db4255286f      2  wakeup   31320.0  2018-11-07T05:30:00-0400
402    08db4255286f      3   onset   44184.0  2018-11-07T23:22:00-0400
...             ...    ...     ...       ...                       ...
11673  d5e47b94477e     21  wakeup  356172.0  2017-11-30T08:41:00-0500
11674  d5e47b94477e     22   onset  366756.0  2017-11-30T23:23:00-0500
11675  d5e47b94477e     22  wakeup  373176.0  2017-12-01T08:18:00-0500
11676  d5e47b94477e     23   onset  383616.0  2017-12-01T22:48:00-0500
11677  d5e47b94477e     23  wakeup  390096.0  2017-12-02T07:48:00-0500

[1562 rows x 5 columns]

In [12]:
drop_train_series=train_series[train_series['series_id'].isin(noNan_user)]
drop_train_series

series_id    step                 timestamp     anglez    enmo
3770820    08db4255286f       0  2018-11-05T10:00:00-0400 -30.845301  0.0447
3770821    08db4255286f       1  2018-11-05T10:00:05-0400 -34.181801  0.0443
3770822    08db4255286f       2  2018-11-05T10:00:10-0400 -33.877102  0.0483
3770823    08db4255286f       3  2018-11-05T10:00:15-0400 -34.282101  0.0680
3770824    08db4255286f       4  2018-11-05T10:00:20-0400 -34.385799  0.0768
...                 ...     ...                       ...        ...     ...
103353655  d5e47b94477e  395815  2017-12-02T15:44:35-0500 -10.064100  0.1552
103353656  d5e47b94477e  395816  2017-12-02T15:44:40-0500 -10.006700  0.1139
103353657  d5e47b94477e  395817  2017-12-02T15:44:45-0500  -3.328600  0.1245
103353658  d5e47b94477e  395818  2017-12-02T15:44:50-0500  -5.903100  0.1363
103353659  d5e47b94477e  395819  2017-12-02T15:44:55-0500 -11.691100  0.0995

[14334840 rows x 5 columns]

### 결측값 데이터 확인하기 2

* 결측값이 전혀 없는 유저 data를 사용하기엔 데이터 수 적당한지가 의문 : 과적합 우려
* train_events에서 결측값인 일부 데이터만 삭제하면?

* train_event의 데이터 수 : TTL 14,508 -> 9,585

In [3]:
import pandas as pd
import pyarrow as pa
import pyarrow.parquet as pq

In [4]:
train_events = pd.read_csv('/Users/soo/Downloads/child/train_events.csv')
new_train_events=train_events.dropna(axis=0).reset_index(drop=True)
new_train_events

series_id  night   event      step                 timestamp
0     038441c925bb      1   onset    4992.0  2018-08-14T22:26:00-0400
1     038441c925bb      1  wakeup   10932.0  2018-08-15T06:41:00-0400
2     038441c925bb      2   onset   20244.0  2018-08-15T19:37:00-0400
3     038441c925bb      2  wakeup   27492.0  2018-08-16T05:41:00-0400
4     038441c925bb      3   onset   39996.0  2018-08-16T23:03:00-0400
...            ...    ...     ...       ...                       ...
9580  fe90110788d2     32  wakeup  547152.0  2017-09-05T09:26:00-0400
9581  fe90110788d2     33   onset  556560.0  2017-09-05T22:30:00-0400
9582  fe90110788d2     33  wakeup  560604.0  2017-09-06T04:07:00-0400
9583  fe90110788d2     34   onset  574620.0  2017-09-06T23:35:00-0400
9584  fe90110788d2     34  wakeup  581604.0  2017-09-07T09:17:00-0400

[9585 rows x 5 columns]

In [6]:
# 데이터 원본
train_series = pq.read_table('/Users/soo/Downloads/child/train_series.parquet').to_pandas()
train_series

series_id    step                 timestamp     anglez    enmo
0          038441c925bb       0  2018-08-14T15:30:00-0400   2.636700  0.0217
1          038441c925bb       1  2018-08-14T15:30:05-0400   2.636800  0.0215
2          038441c925bb       2  2018-08-14T15:30:10-0400   2.637000  0.0216
3          038441c925bb       3  2018-08-14T15:30:15-0400   2.636800  0.0213
4          038441c925bb       4  2018-08-14T15:30:20-0400   2.636800  0.0215
...                 ...     ...                       ...        ...     ...
127946335  fe90110788d2  592375  2017-09-08T00:14:35-0400 -27.277500  0.0204
127946336  fe90110788d2  592376  2017-09-08T00:14:40-0400 -27.032499  0.0233
127946337  fe90110788d2  592377  2017-09-08T00:14:45-0400 -26.841200  0.0202
127946338  fe90110788d2  592378  2017-09-08T00:14:50-0400 -26.723900  0.0199
127946339  fe90110788d2  592379  2017-09-08T00:14:55-0400 -31.521601  0.0205

[127946340 rows x 5 columns]

In [9]:
new_train_series=train_series.drop(['timestamp'],axis=1)

In [10]:
new_train_events_mg = new_train_events.merge(new_train_series, on=('series_id','step'))
new_train_events_mg

series_id  night   event      step                 timestamp  \
0     038441c925bb      1   onset    4992.0  2018-08-14T22:26:00-0400   
1     038441c925bb      1  wakeup   10932.0  2018-08-15T06:41:00-0400   
2     038441c925bb      2   onset   20244.0  2018-08-15T19:37:00-0400   
3     038441c925bb      2  wakeup   27492.0  2018-08-16T05:41:00-0400   
4     038441c925bb      3   onset   39996.0  2018-08-16T23:03:00-0400   
...            ...    ...     ...       ...                       ...   
9580  fe90110788d2     32  wakeup  547152.0  2017-09-05T09:26:00-0400   
9581  fe90110788d2     33   onset  556560.0  2017-09-05T22:30:00-0400   
9582  fe90110788d2     33  wakeup  560604.0  2017-09-06T04:07:00-0400   
9583  fe90110788d2     34   onset  574620.0  2017-09-06T23:35:00-0400   
9584  fe90110788d2     34  wakeup  581604.0  2017-09-07T09:17:00-0400   

         anglez    enmo  
0    -78.690598  0.0099  
1    -61.578201  0.0263  
2     -6.387400  0.0182  
3    -45.355099  0.0165  
4     -1.786700  0.0000  
...         ...     ...  
9580 -56.768902  0.0042  
9581 -16.586700  0.0000  
9582 -43.050900  0.0000  
9583   1.519100  0.0066  
9584  79.053802  0.0000  

[9585 rows x 7 columns]

In [11]:
new_train_events_mg.to_csv("new_train_events_mg.csv", index = True)

In [12]:
new_train_events=new_train_events_mg.copy()

In [13]:
new_train_events['timestamp']

0       2018-08-14T22:26:00-0400
1       2018-08-15T06:41:00-0400
2       2018-08-15T19:37:00-0400
3       2018-08-16T05:41:00-0400
4       2018-08-16T23:03:00-0400
                  ...           
9580    2017-09-05T09:26:00-0400
9581    2017-09-05T22:30:00-0400
9582    2017-09-06T04:07:00-0400
9583    2017-09-06T23:35:00-0400
9584    2017-09-07T09:17:00-0400
Name: timestamp, Length: 9585, dtype: object

In [14]:
from dateutil import parser

In [15]:
new_train_events['timestamp'] = new_train_events['timestamp'].apply(parser.parse)

In [25]:
new_train_events.to_csv("new_train_events.csv", index = False)

In [26]:
#new_train_events : angle, enmo 입력
new_train_events = pd.read_csv('/Users/soo/Downloads/child/new_train_events.csv')

In [27]:
new_train_events

series_id  night   event      step                  timestamp  \
0     038441c925bb      1   onset    4992.0  2018-08-14 22:26:00-04:00   
1     038441c925bb      1  wakeup   10932.0  2018-08-15 06:41:00-04:00   
2     038441c925bb      2   onset   20244.0  2018-08-15 19:37:00-04:00   
3     038441c925bb      2  wakeup   27492.0  2018-08-16 05:41:00-04:00   
4     038441c925bb      3   onset   39996.0  2018-08-16 23:03:00-04:00   
...            ...    ...     ...       ...                        ...   
9580  fe90110788d2     32  wakeup  547152.0  2017-09-05 09:26:00-04:00   
9581  fe90110788d2     33   onset  556560.0  2017-09-05 22:30:00-04:00   
9582  fe90110788d2     33  wakeup  560604.0  2017-09-06 04:07:00-04:00   
9583  fe90110788d2     34   onset  574620.0  2017-09-06 23:35:00-04:00   
9584  fe90110788d2     34  wakeup  581604.0  2017-09-07 09:17:00-04:00   

       anglez    enmo  
0    -78.6906  0.0099  
1    -61.5782  0.0263  
2     -6.3874  0.0182  
3    -45.3551  0.0165  
4     -1.7867  0.0000  
...       ...     ...  
9580 -56.7689  0.0042  
9581 -16.5867  0.0000  
9582 -43.0509  0.0000  
9583   1.5191  0.0066  
9584  79.0538  0.0000  

[9585 rows x 7 columns]

In [28]:
new_train_events.loc[new_train_events['series_id']=='03d92c9f6f8a',:]

series_id  night   event      step                  timestamp   anglez  \
38  03d92c9f6f8a      1   onset    5928.0  2018-05-31 20:14:00-04:00  51.5223   
39  03d92c9f6f8a      1  wakeup   13524.0  2018-06-01 06:47:00-04:00  61.7611   
40  03d92c9f6f8a      2   onset   23220.0  2018-06-01 20:15:00-04:00 -28.8639   
41  03d92c9f6f8a      2  wakeup   30276.0  2018-06-02 06:03:00-04:00 -25.6934   
42  03d92c9f6f8a      3   onset   40668.0  2018-06-02 20:29:00-04:00  19.2442   
43  03d92c9f6f8a      3  wakeup   47952.0  2018-06-03 06:36:00-04:00  78.3559   
44  03d92c9f6f8a      5   onset   75756.0  2018-06-04 21:13:00-04:00  21.6674   
45  03d92c9f6f8a      5  wakeup   82800.0  2018-06-05 07:00:00-04:00 -16.3742   
46  03d92c9f6f8a     11   onset  178464.0  2018-06-10 19:52:00-04:00 -73.0902   
47  03d92c9f6f8a     11  wakeup  186564.0  2018-06-11 07:07:00-04:00 -61.0271   
48  03d92c9f6f8a     12   onset  196260.0  2018-06-11 20:35:00-04:00  -9.6597   
49  03d92c9f6f8a     12  wakeup  203844.0  2018-06-12 07:07:00-04:00 -45.5337   
50  03d92c9f6f8a     14   onset  230820.0  2018-06-13 20:35:00-04:00  40.0117   
51  03d92c9f6f8a     14  wakeup  241872.0  2018-06-14 11:56:00-04:00 -89.3518   
52  03d92c9f6f8a     15   onset  248124.0  2018-06-14 20:37:00-04:00  39.6743   
53  03d92c9f6f8a     15  wakeup  255060.0  2018-06-15 06:15:00-04:00  19.3389   

      enmo  
38  0.0816  
39  0.0000  
40  0.1251  
41  0.0000  
42  0.0000  
43  0.0000  
44  0.0000  
45  0.0000  
46  0.0117  
47  0.0000  
48  0.0000  
49  0.0000  
50  0.0000  
51  0.0001  
52  0.0000  
53  0.0001

In [29]:
new_train_events = new_train_events.set_index(['series_id','night'])
new_train_events

event      step                  timestamp   anglez  \
series_id    night                                                         
038441c925bb 1       onset    4992.0  2018-08-14 22:26:00-04:00 -78.6906   
             1      wakeup   10932.0  2018-08-15 06:41:00-04:00 -61.5782   
             2       onset   20244.0  2018-08-15 19:37:00-04:00  -6.3874   
             2      wakeup   27492.0  2018-08-16 05:41:00-04:00 -45.3551   
             3       onset   39996.0  2018-08-16 23:03:00-04:00  -1.7867   
...                    ...       ...                        ...      ...   
fe90110788d2 32     wakeup  547152.0  2017-09-05 09:26:00-04:00 -56.7689   
             33      onset  556560.0  2017-09-05 22:30:00-04:00 -16.5867   
             33     wakeup  560604.0  2017-09-06 04:07:00-04:00 -43.0509   
             34      onset  574620.0  2017-09-06 23:35:00-04:00   1.5191   
             34     wakeup  581604.0  2017-09-07 09:17:00-04:00  79.0538   

                      enmo  
series_id    night          
038441c925bb 1      0.0099  
             1      0.0263  
             2      0.0182  
             2      0.0165  
             3      0.0000  
...                    ...  
fe90110788d2 32     0.0042  
             33     0.0000  
             33     0.0000  
             34     0.0066  
             34     0.0000  

[9585 rows x 5 columns]

In [30]:
for index, row in new_train_events.iterrows():
    specific_column_value = row['timestamp']  # Replace 'A' with the desired column name
    print(index, specific_column_value)

('038441c925bb', 1) 2018-08-14 22:26:00-04:00
('038441c925bb', 1) 2018-08-15 06:41:00-04:00
('038441c925bb', 2) 2018-08-15 19:37:00-04:00
('038441c925bb', 2) 2018-08-16 05:41:00-04:00
('038441c925bb', 3) 2018-08-16 23:03:00-04:00
('038441c925bb', 3) 2018-08-17 05:10:00-04:00
('038441c925bb', 4) 2018-08-17 23:00:00-04:00
('038441c925bb', 4) 2018-08-18 06:48:00-04:00
('038441c925bb', 6) 2018-08-19 22:18:00-04:00
('038441c925bb', 6) 2018-08-20 07:25:00-04:00
('038441c925bb', 7) 2018-08-20 23:35:00-04:00
('038441c925bb', 7) 2018-08-21 12:07:00-04:00
('038441c925bb', 8) 2018-08-22 00:18:00-04:00
('038441c925bb', 8) 2018-08-22 08:41:00-04:00
('038441c925bb', 10) 2018-08-23 21:41:00-04:00
('038441c925bb', 10) 2018-08-24 08:00:00-04:00
('038441c925bb', 11) 2018-08-24 21:23:00-04:00
('038441c925bb', 11) 2018-08-25 02:37:00-04:00
('038441c925bb', 12) 2018-08-25 21:15:00-04:00
('038441c925bb', 12) 2018-08-26 08:26:00-04:00
('038441c925bb', 13) 2018-08-26 22:22:00-04:00
('038441c925bb', 13) 2018-0

In [15]:
# new_train_events.loc[new_train_events['event']=='onset',new_train_events['event']] = 0
# new_train_events.loc[new_train_events['event']=='wakeup',new_train_events['event']] = 1
# new_train_events

In [33]:
# Convert 'timestamp' column to datetime
new_train_events['timestamp'] = pd.to_datetime(new_train_events['timestamp'])

result_data = [] 
previous_timestamp = None

for index, row in new_train_events.iterrows():
    if previous_timestamp is None:
        previous_timestamp = row['timestamp']
    else:
        if index[0] == previous_index[0] and index[1] == previous_index[1]:
            duration = row['timestamp'] - previous_timestamp
            result_data.append((index[0], index[1], duration))
        else:
            previous_timestamp = row['timestamp']
    previous_index = index

result_df = pd.DataFrame(result_data, columns=['series_id', 'night', 'sleep time'])

result_df

series_id  night      sleep time
0     038441c925bb      1 0 days 08:15:00
1     038441c925bb      2 0 days 10:04:00
2     038441c925bb      3 0 days 06:07:00
3     038441c925bb      4 0 days 07:48:00
4     038441c925bb      6 0 days 09:07:00
...            ...    ...             ...
4785  fe90110788d2     30 0 days 08:34:00
4786  fe90110788d2     31 0 days 08:41:00
4787  fe90110788d2     32 0 days 11:23:00
4788  fe90110788d2     33 0 days 05:37:00
4789  fe90110788d2     34 0 days 09:42:00

[4790 rows x 3 columns]

In [34]:
temp1=result_df.copy()
temp1

series_id  night      sleep time
0     038441c925bb      1 0 days 08:15:00
1     038441c925bb      2 0 days 10:04:00
2     038441c925bb      3 0 days 06:07:00
3     038441c925bb      4 0 days 07:48:00
4     038441c925bb      6 0 days 09:07:00
...            ...    ...             ...
4785  fe90110788d2     30 0 days 08:34:00
4786  fe90110788d2     31 0 days 08:41:00
4787  fe90110788d2     32 0 days 11:23:00
4788  fe90110788d2     33 0 days 05:37:00
4789  fe90110788d2     34 0 days 09:42:00

[4790 rows x 3 columns]

In [42]:
result_user_list=list(temp1['series_id'].unique())
result_user_list

['038441c925bb',
 '03d92c9f6f8a',
 '0402a003dae9',
 '04f547b8017d',
 '05e1944c3818',
 '062cae666e2a',
 '062dbd4c95e6',
 '08db4255286f',
 '0a96f4993bd7',
 '0cd1e3d0ed95',
 '0ce74d6d2106',
 '0cfc06c129cc',
 '0d0ad1e77851',
 '0dee4fda51c3',
 '0ec9fc461819',
 '0ef7d94fde99',
 '0f572d690310',
 '10469f6765bf',
 '1087d7b0ff2e',
 '10f8bc1f7b07',
 '12d01911d509',
 '1319a1935f48',
 '137771d19ca2',
 '137b99e936ab',
 '13b4d6a01d27',
 '148471991ffb',
 '154fe824ed87',
 '16fe2798ed0f',
 '1716cd4163b2',
 '1762ab70ec76',
 '188d4b7cd28b',
 '18a0ca03431d',
 '18b61dd5aae8',
 '1955d568d987',
 '1b92be89db4c',
 '1c7c0bad1263',
 '1d4569cbac0f',
 '1e6717d93c1d',
 '1f96b9668bdf',
 '207eded97727',
 '25e2b3dd9c3b',
 '2654a87be968',
 '27f09a6a858f',
 '280e08693c6d',
 '292a75c0b94e',
 '29c75c018220',
 '29d3469bd15d',
 '2b0a1fa8eba8',
 '2b8d87addea9',
 '2cd2340ca14d',
 '2e9ced2c7976',
 '2f7504d0f426',
 '2fbbee1a38e3',
 '31011ade7c0a',
 '3318a0e3ed6f',
 '33ceeba8918a',
 '3452b878e596',
 '349c5562ee2c',
 '35826366dfc7

In [41]:
result_user_list[0]

'038441c925bb'

In [43]:
for n in result_user_list :
    user_duration_sum = temp1[temp1['series_id'] == 'fe90110788d2']['sleep time'].sum()
user_duration_sum

Timedelta('12 days 07:13:00')

In [214]:
len(temp1.loc[temp1['series_id'] == 'fe90110788d2','duration'])

33

In [44]:
for n in result_user_list :
    user_duration_sum = temp1[temp1['series_id'] == n]['sleep time'].sum() #총 수면 시간
    temp1.loc[temp1['series_id'] == n,'Avg duration'] = user_duration_sum/len(temp1.loc[temp1['series_id'] == n,'sleep time'])
    temp1.loc[temp1['series_id'] == n,'Valid Night Count'] = len(temp1.loc[temp1['series_id'] == n,'sleep time'])
temp1

series_id  night      sleep time              Avg duration  \
0     038441c925bb      1 0 days 08:15:00 0 days 08:46:41.052631578   
1     038441c925bb      2 0 days 10:04:00 0 days 08:46:41.052631578   
2     038441c925bb      3 0 days 06:07:00 0 days 08:46:41.052631578   
3     038441c925bb      4 0 days 07:48:00 0 days 08:46:41.052631578   
4     038441c925bb      6 0 days 09:07:00 0 days 08:46:41.052631578   
...            ...    ...             ...                       ...   
4785  fe90110788d2     30 0 days 08:34:00 0 days 08:56:45.454545454   
4786  fe90110788d2     31 0 days 08:41:00 0 days 08:56:45.454545454   
4787  fe90110788d2     32 0 days 11:23:00 0 days 08:56:45.454545454   
4788  fe90110788d2     33 0 days 05:37:00 0 days 08:56:45.454545454   
4789  fe90110788d2     34 0 days 09:42:00 0 days 08:56:45.454545454   

      Valid Night Count  
0                  19.0  
1                  19.0  
2                  19.0  
3                  19.0  
4                  19.0  
...                 ...  
4785               33.0  
4786               33.0  
4787               33.0  
4788               33.0  
4789               33.0  

[4790 rows x 5 columns]

In [45]:
# for n in result_user_list :
#     user_duration_sum = temp1[temp1['series_id'] == n]['sleep time'].sum() #총 수면 시간
#     temp1.loc[temp1['series_id'] == n,'Avg duration'] = user_duration_sum/len(temp1.loc[temp1['series_id'] == n,'sleep time'])
#     temp1.loc[temp1['series_id'] == n,'Valid Night Count'] = len(temp1.loc[temp1['series_id'] == n,'sleep time'])
# temp1

In [216]:
new_train_events=new_train_events.reset_index(drop=False)
new_train_events

series_id  night   event      step                  timestamp
0     038441c925bb      1   onset    4992.0  2018-08-14 22:26:00-04:00
1     038441c925bb      1  wakeup   10932.0  2018-08-15 06:41:00-04:00
2     038441c925bb      2   onset   20244.0  2018-08-15 19:37:00-04:00
3     038441c925bb      2  wakeup   27492.0  2018-08-16 05:41:00-04:00
4     038441c925bb      3   onset   39996.0  2018-08-16 23:03:00-04:00
...            ...    ...     ...       ...                        ...
9580  fe90110788d2     32  wakeup  547152.0  2017-09-05 09:26:00-04:00
9581  fe90110788d2     33   onset  556560.0  2017-09-05 22:30:00-04:00
9582  fe90110788d2     33  wakeup  560604.0  2017-09-06 04:07:00-04:00
9583  fe90110788d2     34   onset  574620.0  2017-09-06 23:35:00-04:00
9584  fe90110788d2     34  wakeup  581604.0  2017-09-07 09:17:00-04:00

[9585 rows x 5 columns]

In [217]:
# new_train_events['night']

In [46]:
merged_df = new_train_events.merge(temp1, on=('series_id','night'))
merged_df

series_id  night   event      step                  timestamp  \
0     038441c925bb      1   onset    4992.0  2018-08-14 22:26:00-04:00   
1     038441c925bb      1  wakeup   10932.0  2018-08-15 06:41:00-04:00   
2     038441c925bb      2   onset   20244.0  2018-08-15 19:37:00-04:00   
3     038441c925bb      2  wakeup   27492.0  2018-08-16 05:41:00-04:00   
4     038441c925bb      3   onset   39996.0  2018-08-16 23:03:00-04:00   
...            ...    ...     ...       ...                        ...   
9575  fe90110788d2     32  wakeup  547152.0  2017-09-05 09:26:00-04:00   
9576  fe90110788d2     33   onset  556560.0  2017-09-05 22:30:00-04:00   
9577  fe90110788d2     33  wakeup  560604.0  2017-09-06 04:07:00-04:00   
9578  fe90110788d2     34   onset  574620.0  2017-09-06 23:35:00-04:00   
9579  fe90110788d2     34  wakeup  581604.0  2017-09-07 09:17:00-04:00   

       anglez    enmo      sleep time              Avg duration  \
0    -78.6906  0.0099 0 days 08:15:00 0 days 08:46:41.052631578   
1    -61.5782  0.0263 0 days 08:15:00 0 days 08:46:41.052631578   
2     -6.3874  0.0182 0 days 10:04:00 0 days 08:46:41.052631578   
3    -45.3551  0.0165 0 days 10:04:00 0 days 08:46:41.052631578   
4     -1.7867  0.0000 0 days 06:07:00 0 days 08:46:41.052631578   
...       ...     ...             ...                       ...   
9575 -56.7689  0.0042 0 days 11:23:00 0 days 08:56:45.454545454   
9576 -16.5867  0.0000 0 days 05:37:00 0 days 08:56:45.454545454   
9577 -43.0509  0.0000 0 days 05:37:00 0 days 08:56:45.454545454   
9578   1.5191  0.0066 0 days 09:42:00 0 days 08:56:45.454545454   
9579  79.0538  0.0000 0 days 09:42:00 0 days 08:56:45.454545454   

      Valid Night Count  
0                  19.0  
1                  19.0  
2                  19.0  
3                  19.0  
4                  19.0  
...                 ...  
9575               33.0  
9576               33.0  
9577               33.0  
9578               33.0  
9579               33.0  

[9580 rows x 10 columns]

In [219]:
# new_train_events.loc[new_train_events['series_id']=='fbf33b1a2c10',:].head()

In [48]:
len(merged_df['series_id'].unique())

269

In [49]:
# user_list=list(new_train_events['series_id'].unique())
# user_list

In [50]:
merged_df.to_csv("new_train_events_AngleZ, Enmo.csv", index = False)

In [51]:
new_train_events = merged_df.set_index(['series_id','Valid Night Count','Avg duration','sleep time','night'])
new_train_events

event  \
series_id    Valid Night Count Avg duration              sleep time      night           
038441c925bb 19.0              0 days 08:46:41.052631578 0 days 08:15:00 1       onset   
                                                                         1      wakeup   
                                                         0 days 10:04:00 2       onset   
                                                                         2      wakeup   
                                                         0 days 06:07:00 3       onset   
...                                                                                ...   
fe90110788d2 33.0              0 days 08:56:45.454545454 0 days 11:23:00 32     wakeup   
                                                         0 days 05:37:00 33      onset   
                                                                         33     wakeup   
                                                         0 days 09:42:00 34      onset   
                                                                         34     wakeup   

                                                                                    step  \
series_id    Valid Night Count Avg duration              sleep time      night             
038441c925bb 19.0              0 days 08:46:41.052631578 0 days 08:15:00 1        4992.0   
                                                                         1       10932.0   
                                                         0 days 10:04:00 2       20244.0   
                                                                         2       27492.0   
                                                         0 days 06:07:00 3       39996.0   
...                                                                                  ...   
fe90110788d2 33.0              0 days 08:56:45.454545454 0 days 11:23:00 32     547152.0   
                                                         0 days 05:37:00 33     556560.0   
                                                                         33     560604.0   
                                                         0 days 09:42:00 34     574620.0   
                                                                         34     581604.0   

                                                                                                timestamp  \
series_id    Valid Night Count Avg duration              sleep time      night                              
038441c925bb 19.0              0 days 08:46:41.052631578 0 days 08:15:00 1      2018-08-14 22:26:00-04:00   
                                                                         1      2018-08-15 06:41:00-04:00   
                                                         0 days 10:04:00 2      2018-08-15 19:37:00-04:00   
                                                                         2      2018-08-16 05:41:00-04:00   
                                                         0 days 06:07:00 3      2018-08-16 23:03:00-04:00   
...                                                                                                   ...   
fe90110788d2 33.0              0 days 08:56:45.454545454 0 days 11:23:00 32     2017-09-05 09:26:00-04:00   
                                                         0 days 05:37:00 33     2017-09-05 22:30:00-04:00   
                                                                         33     2017-09-06 04:07:00-04:00   
                                                         0 days 09:42:00 34     2017-09-06 23:35:00-04:00   
                                                                         34     2017-09-07 09:17:00-04:00   

                                                                                 anglez  \
series_id    Valid Night Count Avg duration              sleep time      night            
038441c925bb 19.0              0 days 08:46:41.052631578 0 days 08:15:00 1     -78.6906   
                      

In [52]:
new_train_events.to_csv("new_train_events_AngleZ, Enmo_MultiColumn.csv", index = True)

In [53]:
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [54]:
#전체 평균 수면시간 이상/미만 나누기
new_train_events_allAVG=new_train_events.copy()

In [56]:
new_train_events

event  \
series_id    Valid Night Count Avg duration              sleep time      night           
038441c925bb 19.0              0 days 08:46:41.052631578 0 days 08:15:00 1       onset   
                                                                         1      wakeup   
                                                         0 days 10:04:00 2       onset   
                                                                         2      wakeup   
                                                         0 days 06:07:00 3       onset   
...                                                                                ...   
fe90110788d2 33.0              0 days 08:56:45.454545454 0 days 11:23:00 32     wakeup   
                                                         0 days 05:37:00 33      onset   
                                                                         33     wakeup   
                                                         0 days 09:42:00 34      onset   
                                                                         34     wakeup   

                                                                                    step  \
series_id    Valid Night Count Avg duration              sleep time      night             
038441c925bb 19.0              0 days 08:46:41.052631578 0 days 08:15:00 1        4992.0   
                                                                         1       10932.0   
                                                         0 days 10:04:00 2       20244.0   
                                                                         2       27492.0   
                                                         0 days 06:07:00 3       39996.0   
...                                                                                  ...   
fe90110788d2 33.0              0 days 08:56:45.454545454 0 days 11:23:00 32     547152.0   
                                                         0 days 05:37:00 33     556560.0   
                                                                         33     560604.0   
                                                         0 days 09:42:00 34     574620.0   
                                                                         34     581604.0   

                                                                                                timestamp  \
series_id    Valid Night Count Avg duration              sleep time      night                              
038441c925bb 19.0              0 days 08:46:41.052631578 0 days 08:15:00 1      2018-08-14 22:26:00-04:00   
                                                                         1      2018-08-15 06:41:00-04:00   
                                                         0 days 10:04:00 2      2018-08-15 19:37:00-04:00   
                                                                         2      2018-08-16 05:41:00-04:00   
                                                         0 days 06:07:00 3      2018-08-16 23:03:00-04:00   
...                                                                                                   ...   
fe90110788d2 33.0              0 days 08:56:45.454545454 0 days 11:23:00 32     2017-09-05 09:26:00-04:00   
                                                         0 days 05:37:00 33     2017-09-05 22:30:00-04:00   
                                                                         33     2017-09-06 04:07:00-04:00   
                                                         0 days 09:42:00 34     2017-09-06 23:35:00-04:00   
                                                                         34     2017-09-07 09:17:00-04:00   

                                                                                 anglez  \
series_id    Valid Night Count Avg duration              sleep time      night            
038441c925bb 19.0              0 days 08:46:41.052631578 0 days 08:15:00 1     -78.6906   
                      

In [57]:
new_train_events_allAVG=new_train_events_allAVG.reset_index()
new_train_events_allAVG

series_id  Valid Night Count              Avg duration  \
0     038441c925bb               19.0 0 days 08:46:41.052631578   
1     038441c925bb               19.0 0 days 08:46:41.052631578   
2     038441c925bb               19.0 0 days 08:46:41.052631578   
3     038441c925bb               19.0 0 days 08:46:41.052631578   
4     038441c925bb               19.0 0 days 08:46:41.052631578   
...            ...                ...                       ...   
9575  fe90110788d2               33.0 0 days 08:56:45.454545454   
9576  fe90110788d2               33.0 0 days 08:56:45.454545454   
9577  fe90110788d2               33.0 0 days 08:56:45.454545454   
9578  fe90110788d2               33.0 0 days 08:56:45.454545454   
9579  fe90110788d2               33.0 0 days 08:56:45.454545454   

          sleep time  night   event      step                  timestamp  \
0    0 days 08:15:00      1   onset    4992.0  2018-08-14 22:26:00-04:00   
1    0 days 08:15:00      1  wakeup   10932.0  2018-08-15 06:41:00-04:00   
2    0 days 10:04:00      2   onset   20244.0  2018-08-15 19:37:00-04:00   
3    0 days 10:04:00      2  wakeup   27492.0  2018-08-16 05:41:00-04:00   
4    0 days 06:07:00      3   onset   39996.0  2018-08-16 23:03:00-04:00   
...              ...    ...     ...       ...                        ...   
9575 0 days 11:23:00     32  wakeup  547152.0  2017-09-05 09:26:00-04:00   
9576 0 days 05:37:00     33   onset  556560.0  2017-09-05 22:30:00-04:00   
9577 0 days 05:37:00     33  wakeup  560604.0  2017-09-06 04:07:00-04:00   
9578 0 days 09:42:00     34   onset  574620.0  2017-09-06 23:35:00-04:00   
9579 0 days 09:42:00     34  wakeup  581604.0  2017-09-07 09:17:00-04:00   

       anglez    enmo  
0    -78.6906  0.0099  
1    -61.5782  0.0263  
2     -6.3874  0.0182  
3    -45.3551  0.0165  
4     -1.7867  0.0000  
...       ...     ...  
9575 -56.7689  0.0042  
9576 -16.5867  0.0000  
9577 -43.0509  0.0000  
9578   1.5191  0.0066  
9579  79.0538  0.0000  

[9580 rows x 10 columns]

In [84]:
new_train_events_allAVG.rename(columns={'Avg duration': 'Avg sleep time'},inplace=True)
new_train_events_allAVG

series_id  Valid Night Count            Avg sleep time  \
0     038441c925bb               19.0 0 days 08:46:41.052631578   
1     038441c925bb               19.0 0 days 08:46:41.052631578   
2     038441c925bb               19.0 0 days 08:46:41.052631578   
3     038441c925bb               19.0 0 days 08:46:41.052631578   
4     038441c925bb               19.0 0 days 08:46:41.052631578   
...            ...                ...                       ...   
9575  fe90110788d2               33.0 0 days 08:56:45.454545454   
9576  fe90110788d2               33.0 0 days 08:56:45.454545454   
9577  fe90110788d2               33.0 0 days 08:56:45.454545454   
9578  fe90110788d2               33.0 0 days 08:56:45.454545454   
9579  fe90110788d2               33.0 0 days 08:56:45.454545454   

          sleep time  night   event      step                  timestamp  \
0    0 days 08:15:00      1   onset    4992.0  2018-08-14 22:26:00-04:00   
1    0 days 08:15:00      1  wakeup   10932.0  2018-08-15 06:41:00-04:00   
2    0 days 10:04:00      2   onset   20244.0  2018-08-15 19:37:00-04:00   
3    0 days 10:04:00      2  wakeup   27492.0  2018-08-16 05:41:00-04:00   
4    0 days 06:07:00      3   onset   39996.0  2018-08-16 23:03:00-04:00   
...              ...    ...     ...       ...                        ...   
9575 0 days 11:23:00     32  wakeup  547152.0  2017-09-05 09:26:00-04:00   
9576 0 days 05:37:00     33   onset  556560.0  2017-09-05 22:30:00-04:00   
9577 0 days 05:37:00     33  wakeup  560604.0  2017-09-06 04:07:00-04:00   
9578 0 days 09:42:00     34   onset  574620.0  2017-09-06 23:35:00-04:00   
9579 0 days 09:42:00     34  wakeup  581604.0  2017-09-07 09:17:00-04:00   

       anglez    enmo  
0    -78.6906  0.0099  
1    -61.5782  0.0263  
2     -6.3874  0.0182  
3    -45.3551  0.0165  
4     -1.7867  0.0000  
...       ...     ...  
9575 -56.7689  0.0042  
9576 -16.5867  0.0000  
9577 -43.0509  0.0000  
9578   1.5191  0.0066  
9579  79.0538  0.0000  

[9580 rows x 10 columns]

In [86]:
user_list=list(new_train_events_allAVG['series_id'].unique())

In [87]:
len(user_list)

269

In [88]:
new_train_events_allAVG['Avg sleep time'].sum()

Timedelta('3453 days 11:53:59.999997440')

In [95]:
temp_user_avgtime=new_train_events_allAVG.drop_duplicates(['series_id'], keep='first', inplace=False, ignore_index=True)
temp_user_avgtime

series_id  Valid Night Count            Avg sleep time  \
0    038441c925bb               19.0 0 days 08:46:41.052631578   
1    03d92c9f6f8a                8.0    0 days 10:52:37.500000   
2    0402a003dae9               18.0 0 days 06:33:46.666666666   
3    04f547b8017d               21.0 0 days 10:00:45.714285714   
4    05e1944c3818                7.0           0 days 08:27:00   
..            ...                ...                       ...   
264  fa149c3c4bde               16.0    0 days 09:50:52.500000   
265  fb223ed2278c                5.0           0 days 09:47:48   
266  fbf33b1a2c10               20.0           0 days 07:32:33   
267  fcca183903b7               35.0 0 days 08:30:22.285714285   
268  fe90110788d2               33.0 0 days 08:56:45.454545454   

         sleep time  night  event     step                  timestamp  \
0   0 days 08:15:00      1  onset   4992.0  2018-08-14 22:26:00-04:00   
1   0 days 10:33:00      1  onset   5928.0  2018-05-31 20:14:00-04:00   
2   0 days 06:22:00      1  onset   8364.0  2018-12-19 00:22:00-05:00   
3   0 days 10:10:00      2  onset  23484.0  2018-11-29 20:37:00-05:00   
4   0 days 09:30:00      2  onset  20520.0  2018-11-17 22:30:00-05:00   
..              ...    ...    ...      ...                        ...   
264 0 days 08:37:00      1  onset   3012.0  2018-08-30 23:11:00-04:00   
265 0 days 09:50:00      1  onset   7536.0  2018-12-14 22:28:00-05:00   
266 0 days 06:55:00      2  onset  23412.0  2018-03-31 00:46:00-04:00   
267 0 days 08:07:00      1  onset   6300.0  2019-03-26 21:45:00-04:00   
268 0 days 09:27:00      2  onset  21048.0  2017-08-05 22:44:00-04:00   

      anglez    enmo  
0   -78.6906  0.0099  
1    51.5223  0.0816  
2    29.0663  0.0000  
3    15.0465  0.0014  
4    33.2058  0.0000  
..       ...     ...  
264  60.4461  0.0116  
265  13.3682  0.0065  
266   2.8057  0.0012  
267 -18.7579  0.1039  
268  48.0831  0.0505  

[269 rows x 10 columns]

In [96]:
temp_user_avgtime.drop(['Valid Night Count','sleep time','night','event','step','timestamp','anglez','enmo'],axis=1,inplace=True)
temp_user_avgtime

/var/folders/gm/w1dhc0qj1x72sgdrxk_zc4jw0000gn/T/ipykernel_11487/3372747863.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  temp_user_avgtime.drop(['Valid Night Count','sleep time','night','event','step','timestamp','anglez','enmo'],axis=1,inplace=True)


series_id            Avg sleep time
0    038441c925bb 0 days 08:46:41.052631578
1    03d92c9f6f8a    0 days 10:52:37.500000
2    0402a003dae9 0 days 06:33:46.666666666
3    04f547b8017d 0 days 10:00:45.714285714
4    05e1944c3818           0 days 08:27:00
..            ...                       ...
264  fa149c3c4bde    0 days 09:50:52.500000
265  fb223ed2278c           0 days 09:47:48
266  fbf33b1a2c10           0 days 07:32:33
267  fcca183903b7 0 days 08:30:22.285714285
268  fe90110788d2 0 days 08:56:45.454545454

[269 rows x 2 columns]

In [93]:
len(temp_user_avgtime['series_id'])

269

In [97]:
temp_user_avgtime['Avg sleep time'].sum()

Timedelta('96 days 15:57:15.977310998')

In [98]:
ttl_avg_sleeptime=temp_user_avgtime['Avg sleep time'].sum()/len(temp_user_avgtime['series_id'])
ttl_avg_sleeptime

Timedelta('0 days 08:37:27.717387773')

In [99]:
new_train_events_allAVG['AllUserAvgSleep']=ttl_avg_sleeptime
new_train_events_allAVG

series_id  Valid Night Count            Avg sleep time  \
0     038441c925bb               19.0 0 days 08:46:41.052631578   
1     038441c925bb               19.0 0 days 08:46:41.052631578   
2     038441c925bb               19.0 0 days 08:46:41.052631578   
3     038441c925bb               19.0 0 days 08:46:41.052631578   
4     038441c925bb               19.0 0 days 08:46:41.052631578   
...            ...                ...                       ...   
9575  fe90110788d2               33.0 0 days 08:56:45.454545454   
9576  fe90110788d2               33.0 0 days 08:56:45.454545454   
9577  fe90110788d2               33.0 0 days 08:56:45.454545454   
9578  fe90110788d2               33.0 0 days 08:56:45.454545454   
9579  fe90110788d2               33.0 0 days 08:56:45.454545454   

          sleep time  night   event      step                  timestamp  \
0    0 days 08:15:00      1   onset    4992.0  2018-08-14 22:26:00-04:00   
1    0 days 08:15:00      1  wakeup   10932.0  2018-08-15 06:41:00-04:00   
2    0 days 10:04:00      2   onset   20244.0  2018-08-15 19:37:00-04:00   
3    0 days 10:04:00      2  wakeup   27492.0  2018-08-16 05:41:00-04:00   
4    0 days 06:07:00      3   onset   39996.0  2018-08-16 23:03:00-04:00   
...              ...    ...     ...       ...                        ...   
9575 0 days 11:23:00     32  wakeup  547152.0  2017-09-05 09:26:00-04:00   
9576 0 days 05:37:00     33   onset  556560.0  2017-09-05 22:30:00-04:00   
9577 0 days 05:37:00     33  wakeup  560604.0  2017-09-06 04:07:00-04:00   
9578 0 days 09:42:00     34   onset  574620.0  2017-09-06 23:35:00-04:00   
9579 0 days 09:42:00     34  wakeup  581604.0  2017-09-07 09:17:00-04:00   

       anglez    enmo           AllUserAvgSleep  
0    -78.6906  0.0099 0 days 08:37:27.717387773  
1    -61.5782  0.0263 0 days 08:37:27.717387773  
2     -6.3874  0.0182 0 days 08:37:27.717387773  
3    -45.3551  0.0165 0 days 08:37:27.717387773  
4     -1.7867  0.0000 0 days 08:37:27.717387773  
...       ...     ...                       ...  
9575 -56.7689  0.0042 0 days 08:37:27.717387773  
9576 -16.5867  0.0000 0 days 08:37:27.717387773  
9577 -43.0509  0.0000 0 days 08:37:27.717387773  
9578   1.5191  0.0066 0 days 08:37:27.717387773  
9579  79.0538  0.0000 0 days 08:37:27.717387773  

[9580 rows x 11 columns]

In [102]:
new_train_events_allAVG.loc[0,'Avg sleep time']

Timedelta('0 days 08:46:41.052631578')

In [103]:
new_train_events_allAVG.loc[0,'AllUserAvgSleep']

Timedelta('0 days 08:37:27.717387773')

In [122]:
temp_new_train_events_allAVG=new_train_events_allAVG.copy()

In [128]:
#전체 평균 수면시간 보다 이상이면 1, 미만이면 0
def update_above_auas(row):
    if row['Avg sleep time'] >= row['AllUserAvgSleep']:
        return 1
    else:
        return 0

temp_new_train_events_allAVG['AboveAUAS'] = temp_new_train_events_allAVG.apply(update_above_auas, axis=1)
temp_new_train_events_allAVG

series_id  Valid Night Count            Avg sleep time  \
0     038441c925bb               19.0 0 days 08:46:41.052631578   
1     038441c925bb               19.0 0 days 08:46:41.052631578   
2     038441c925bb               19.0 0 days 08:46:41.052631578   
3     038441c925bb               19.0 0 days 08:46:41.052631578   
4     038441c925bb               19.0 0 days 08:46:41.052631578   
...            ...                ...                       ...   
9575  fe90110788d2               33.0 0 days 08:56:45.454545454   
9576  fe90110788d2               33.0 0 days 08:56:45.454545454   
9577  fe90110788d2               33.0 0 days 08:56:45.454545454   
9578  fe90110788d2               33.0 0 days 08:56:45.454545454   
9579  fe90110788d2               33.0 0 days 08:56:45.454545454   

          sleep time  night   event      step                  timestamp  \
0    0 days 08:15:00      1   onset    4992.0  2018-08-14 22:26:00-04:00   
1    0 days 08:15:00      1  wakeup   10932.0  2018-08-15 06:41:00-04:00   
2    0 days 10:04:00      2   onset   20244.0  2018-08-15 19:37:00-04:00   
3    0 days 10:04:00      2  wakeup   27492.0  2018-08-16 05:41:00-04:00   
4    0 days 06:07:00      3   onset   39996.0  2018-08-16 23:03:00-04:00   
...              ...    ...     ...       ...                        ...   
9575 0 days 11:23:00     32  wakeup  547152.0  2017-09-05 09:26:00-04:00   
9576 0 days 05:37:00     33   onset  556560.0  2017-09-05 22:30:00-04:00   
9577 0 days 05:37:00     33  wakeup  560604.0  2017-09-06 04:07:00-04:00   
9578 0 days 09:42:00     34   onset  574620.0  2017-09-06 23:35:00-04:00   
9579 0 days 09:42:00     34  wakeup  581604.0  2017-09-07 09:17:00-04:00   

       anglez    enmo           AllUserAvgSleep  AboveAUAS  
0    -78.6906  0.0099 0 days 08:37:27.717387773          1  
1    -61.5782  0.0263 0 days 08:37:27.717387773          1  
2     -6.3874  0.0182 0 days 08:37:27.717387773          1  
3    -45.3551  0.0165 0 days 08:37:27.717387773          1  
4     -1.7867  0.0000 0 days 08:37:27.717387773          1  
...       ...     ...                       ...        ...  
9575 -56.7689  0.0042 0 days 08:37:27.717387773          1  
9576 -16.5867  0.0000 0 days 08:37:27.717387773          1  
9577 -43.0509  0.0000 0 days 08:37:27.717387773          1  
9578   1.5191  0.0066 0 days 08:37:27.717387773          1  
9579  79.0538  0.0000 0 days 08:37:27.717387773          1  

[9580 rows x 12 columns]

In [131]:
#이벤트 수면 시작 : 1 ,  수면 종료 : 0
def event_invert(row):
    if row['event'] == 'onset':
        return 1
    else:
        return 0

temp_new_train_events_allAVG['event'] = temp_new_train_events_allAVG.apply(event_invert, axis=1)
temp_new_train_events_allAVG

series_id  Valid Night Count            Avg sleep time  \
0     038441c925bb               19.0 0 days 08:46:41.052631578   
1     038441c925bb               19.0 0 days 08:46:41.052631578   
2     038441c925bb               19.0 0 days 08:46:41.052631578   
3     038441c925bb               19.0 0 days 08:46:41.052631578   
4     038441c925bb               19.0 0 days 08:46:41.052631578   
...            ...                ...                       ...   
9575  fe90110788d2               33.0 0 days 08:56:45.454545454   
9576  fe90110788d2               33.0 0 days 08:56:45.454545454   
9577  fe90110788d2               33.0 0 days 08:56:45.454545454   
9578  fe90110788d2               33.0 0 days 08:56:45.454545454   
9579  fe90110788d2               33.0 0 days 08:56:45.454545454   

          sleep time  night  event      step                  timestamp  \
0    0 days 08:15:00      1      1    4992.0  2018-08-14 22:26:00-04:00   
1    0 days 08:15:00      1      0   10932.0  2018-08-15 06:41:00-04:00   
2    0 days 10:04:00      2      1   20244.0  2018-08-15 19:37:00-04:00   
3    0 days 10:04:00      2      0   27492.0  2018-08-16 05:41:00-04:00   
4    0 days 06:07:00      3      1   39996.0  2018-08-16 23:03:00-04:00   
...              ...    ...    ...       ...                        ...   
9575 0 days 11:23:00     32      0  547152.0  2017-09-05 09:26:00-04:00   
9576 0 days 05:37:00     33      1  556560.0  2017-09-05 22:30:00-04:00   
9577 0 days 05:37:00     33      0  560604.0  2017-09-06 04:07:00-04:00   
9578 0 days 09:42:00     34      1  574620.0  2017-09-06 23:35:00-04:00   
9579 0 days 09:42:00     34      0  581604.0  2017-09-07 09:17:00-04:00   

       anglez    enmo           AllUserAvgSleep  AboveAUAS  
0    -78.6906  0.0099 0 days 08:37:27.717387773          1  
1    -61.5782  0.0263 0 days 08:37:27.717387773          1  
2     -6.3874  0.0182 0 days 08:37:27.717387773          1  
3    -45.3551  0.0165 0 days 08:37:27.717387773          1  
4     -1.7867  0.0000 0 days 08:37:27.717387773          1  
...       ...     ...                       ...        ...  
9575 -56.7689  0.0042 0 days 08:37:27.717387773          1  
9576 -16.5867  0.0000 0 days 08:37:27.717387773          1  
9577 -43.0509  0.0000 0 days 08:37:27.717387773          1  
9578   1.5191  0.0066 0 days 08:37:27.717387773          1  
9579  79.0538  0.0000 0 days 08:37:27.717387773          1  

[9580 rows x 12 columns]

In [132]:
temp_new_train_events_allAVG[temp_new_train_events_allAVG['AboveAUAS']==0]

series_id  Valid Night Count            Avg sleep time  \
54    0402a003dae9               18.0 0 days 06:33:46.666666666   
55    0402a003dae9               18.0 0 days 06:33:46.666666666   
56    0402a003dae9               18.0 0 days 06:33:46.666666666   
57    0402a003dae9               18.0 0 days 06:33:46.666666666   
58    0402a003dae9               18.0 0 days 06:33:46.666666666   
...            ...                ...                       ...   
9509  fcca183903b7               35.0 0 days 08:30:22.285714285   
9510  fcca183903b7               35.0 0 days 08:30:22.285714285   
9511  fcca183903b7               35.0 0 days 08:30:22.285714285   
9512  fcca183903b7               35.0 0 days 08:30:22.285714285   
9513  fcca183903b7               35.0 0 days 08:30:22.285714285   

          sleep time  night  event      step                  timestamp  \
54   0 days 06:22:00      1      1    8364.0  2018-12-19 00:22:00-05:00   
55   0 days 06:22:00      1      0   12948.0  2018-12-19 06:44:00-05:00   
56   0 days 07:44:00      2      1   24396.0  2018-12-19 22:38:00-05:00   
57   0 days 07:44:00      2      0   29964.0  2018-12-20 06:22:00-05:00   
58   0 days 02:59:00      6      1   94032.0  2018-12-23 23:21:00-05:00   
...              ...    ...    ...       ...                        ...   
9509 0 days 08:24:00     33      0  565824.0  2019-04-28 06:52:00-04:00   
9510 0 days 09:19:00     34      1  577344.0  2019-04-28 22:52:00-04:00   
9511 0 days 09:19:00     34      0  584052.0  2019-04-29 08:11:00-04:00   
9512 0 days 09:26:00     35      1  595344.0  2019-04-29 23:52:00-04:00   
9513 0 days 09:26:00     35      0  602136.0  2019-04-30 09:18:00-04:00   

       anglez    enmo           AllUserAvgSleep  AboveAUAS  
54    29.0663  0.0000 0 days 08:37:27.717387773          0  
55    42.9404  0.0008 0 days 08:37:27.717387773          0  
56     1.8807  0.0099 0 days 08:37:27.717387773          0  
57   -39.7638  0.0507 0 days 08:37:27.717387773          0  
58   -77.3549  0.0303 0 days 08:37:27.717387773          0  
...       ...     ...                       ...        ...  
9509  -7.3872  0.0033 0 days 08:37:27.717387773          0  
9510 -44.7628  0.0000 0 days 08:37:27.717387773          0  
9511 -68.2084  0.0341 0 days 08:37:27.717387773          0  
9512 -12.1260  0.0517 0 days 08:37:27.717387773          0  
9513  82.3611  0.0000 0 days 08:37:27.717387773          0  

[4136 rows x 12 columns]

In [133]:
temp_new_train_events_allAVG[temp_new_train_events_allAVG['AboveAUAS']==1]

series_id  Valid Night Count            Avg sleep time  \
0     038441c925bb               19.0 0 days 08:46:41.052631578   
1     038441c925bb               19.0 0 days 08:46:41.052631578   
2     038441c925bb               19.0 0 days 08:46:41.052631578   
3     038441c925bb               19.0 0 days 08:46:41.052631578   
4     038441c925bb               19.0 0 days 08:46:41.052631578   
...            ...                ...                       ...   
9575  fe90110788d2               33.0 0 days 08:56:45.454545454   
9576  fe90110788d2               33.0 0 days 08:56:45.454545454   
9577  fe90110788d2               33.0 0 days 08:56:45.454545454   
9578  fe90110788d2               33.0 0 days 08:56:45.454545454   
9579  fe90110788d2               33.0 0 days 08:56:45.454545454   

          sleep time  night  event      step                  timestamp  \
0    0 days 08:15:00      1      1    4992.0  2018-08-14 22:26:00-04:00   
1    0 days 08:15:00      1      0   10932.0  2018-08-15 06:41:00-04:00   
2    0 days 10:04:00      2      1   20244.0  2018-08-15 19:37:00-04:00   
3    0 days 10:04:00      2      0   27492.0  2018-08-16 05:41:00-04:00   
4    0 days 06:07:00      3      1   39996.0  2018-08-16 23:03:00-04:00   
...              ...    ...    ...       ...                        ...   
9575 0 days 11:23:00     32      0  547152.0  2017-09-05 09:26:00-04:00   
9576 0 days 05:37:00     33      1  556560.0  2017-09-05 22:30:00-04:00   
9577 0 days 05:37:00     33      0  560604.0  2017-09-06 04:07:00-04:00   
9578 0 days 09:42:00     34      1  574620.0  2017-09-06 23:35:00-04:00   
9579 0 days 09:42:00     34      0  581604.0  2017-09-07 09:17:00-04:00   

       anglez    enmo           AllUserAvgSleep  AboveAUAS  
0    -78.6906  0.0099 0 days 08:37:27.717387773          1  
1    -61.5782  0.0263 0 days 08:37:27.717387773          1  
2     -6.3874  0.0182 0 days 08:37:27.717387773          1  
3    -45.3551  0.0165 0 days 08:37:27.717387773          1  
4     -1.7867  0.0000 0 days 08:37:27.717387773          1  
...       ...     ...                       ...        ...  
9575 -56.7689  0.0042 0 days 08:37:27.717387773          1  
9576 -16.5867  0.0000 0 days 08:37:27.717387773          1  
9577 -43.0509  0.0000 0 days 08:37:27.717387773          1  
9578   1.5191  0.0066 0 days 08:37:27.717387773          1  
9579  79.0538  0.0000 0 days 08:37:27.717387773          1  

[5444 rows x 12 columns]

In [173]:
new_train_events = temp_new_train_events_allAVG.set_index(['AllUserAvgSleep','series_id','Valid Night Count','Avg sleep time','AboveAUAS','sleep time','night'])


In [175]:
new_train_events

event  \
AllUserAvgSleep           series_id    Valid Night Count Avg sleep time            AboveAUAS sleep time      night          
0 days 08:37:27.717387773 038441c925bb 19.0              0 days 08:46:41.052631578 1         0 days 08:15:00 1          1   
                                                                                                             1          0   
                                                                                             0 days 10:04:00 2          1   
                                                                                                             2          0   
                                                                                             0 days 06:07:00 3          1   
...                                                                                                                   ...   
                          fe90110788d2 33.0              0 days 08:56:45.454545454 1         0 days 11:23:00 32         0   
                                                                                             0 days 05:37:00 33         1   
                                                                                                             33         0   
                                                                                             0 days 09:42:00 34         1   
                                                                                                             34         0   

                                                                                                                        step  \
AllUserAvgSleep           series_id    Valid Night Count Avg sleep time            AboveAUAS sleep time      night             
0 days 08:37:27.717387773 038441c925bb 19.0              0 days 08:46:41.052631578 1         0 days 08:15:00 1        4992.0   
                                                                                                             1       10932.0   
                                                                                             0 days 10:04:00 2       20244.0   
                                                                                                             2       27492.0   
                                                                                             0 days 06:07:00 3       39996.0   
...                                                                                                                      ...   
                          fe90110788d2 33.0              0 days 08:56:45.454545454 1         0 days 11:23:00 32     547152.0   
                                                                                             0 days 05:37:00 33     556560.0   
                                                                                                             33     560604.0   
                                                                                             0 days 09:42:00 34     574620.0   
                                                                                                             34     581604.0   

                                                                                                                                    timestamp  \
AllUserAvgSleep           series_id    Valid Night Count Avg sleep time            AboveAUAS sleep time      night                              
0 days 08:37:27.717387773 038441c925bb 19.0              0 days 08:46:41.052631578 1         0 days 08:15:00 1      2018-08-14 22:26:00-04:00   
                                                                                                             1      2018-08-15 06:41:00-04:00   
                                                                                             0 days 10:04:00 2      2018-08-15 19:37:00-04:00   
                                                                                                    

In [179]:
forSaveReIndex=new_train_events.reset_index(drop=False,inplace=False)

In [180]:
forSaveReIndex.to_csv('new_train_events_2.csv', index = False)

### overall 자료

In [184]:
allUserAnalysis=temp_new_train_events_allAVG.drop(['Avg sleep time','sleep time','night','event','step','timestamp','anglez','enmo'],axis=1)
allUserAnalysis

series_id  Valid Night Count           AllUserAvgSleep  AboveAUAS
0     038441c925bb               19.0 0 days 08:37:27.717387773          1
1     038441c925bb               19.0 0 days 08:37:27.717387773          1
2     038441c925bb               19.0 0 days 08:37:27.717387773          1
3     038441c925bb               19.0 0 days 08:37:27.717387773          1
4     038441c925bb               19.0 0 days 08:37:27.717387773          1
...            ...                ...                       ...        ...
9575  fe90110788d2               33.0 0 days 08:37:27.717387773          1
9576  fe90110788d2               33.0 0 days 08:37:27.717387773          1
9577  fe90110788d2               33.0 0 days 08:37:27.717387773          1
9578  fe90110788d2               33.0 0 days 08:37:27.717387773          1
9579  fe90110788d2               33.0 0 days 08:37:27.717387773          1

[9580 rows x 4 columns]

In [159]:
allUserAnalysis=allUserAnalysis.drop_duplicates(['series_id'], keep='first', inplace=False, ignore_index=True)
allUserAnalysis

series_id  Valid Night Count           AllUserAvgSleep  AboveAUAS
0    038441c925bb               19.0 0 days 08:37:27.717387773          1
1    03d92c9f6f8a                8.0 0 days 08:37:27.717387773          1
2    0402a003dae9               18.0 0 days 08:37:27.717387773          0
3    04f547b8017d               21.0 0 days 08:37:27.717387773          1
4    05e1944c3818                7.0 0 days 08:37:27.717387773          0
..            ...                ...                       ...        ...
264  fa149c3c4bde               16.0 0 days 08:37:27.717387773          1
265  fb223ed2278c                5.0 0 days 08:37:27.717387773          1
266  fbf33b1a2c10               20.0 0 days 08:37:27.717387773          0
267  fcca183903b7               35.0 0 days 08:37:27.717387773          0
268  fe90110788d2               33.0 0 days 08:37:27.717387773          1

[269 rows x 4 columns]

In [160]:
allUserAnalysis['Valid Night Count'].sum()/len(allUserAnalysis['Valid Night Count'])

17.806691449814128

In [161]:
allUserAnalysis['Valid Night Count']=allUserAnalysis['Valid Night Count'].sum()/len(allUserAnalysis['Valid Night Count'])

/var/folders/gm/w1dhc0qj1x72sgdrxk_zc4jw0000gn/T/ipykernel_11487/3697094727.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  allUserAnalysis['Valid Night Count']=allUserAnalysis['Valid Night Count'].sum()/len(allUserAnalysis['Valid Night Count'])


In [162]:
allUserAnalysis

series_id  Valid Night Count           AllUserAvgSleep  AboveAUAS
0    038441c925bb          17.806691 0 days 08:37:27.717387773          1
1    03d92c9f6f8a          17.806691 0 days 08:37:27.717387773          1
2    0402a003dae9          17.806691 0 days 08:37:27.717387773          0
3    04f547b8017d          17.806691 0 days 08:37:27.717387773          1
4    05e1944c3818          17.806691 0 days 08:37:27.717387773          0
..            ...                ...                       ...        ...
264  fa149c3c4bde          17.806691 0 days 08:37:27.717387773          1
265  fb223ed2278c          17.806691 0 days 08:37:27.717387773          1
266  fbf33b1a2c10          17.806691 0 days 08:37:27.717387773          0
267  fcca183903b7          17.806691 0 days 08:37:27.717387773          0
268  fe90110788d2          17.806691 0 days 08:37:27.717387773          1

[269 rows x 4 columns]

In [163]:
allUserAnalysis['AboveAUAS : 1'] = len(allUserAnalysis[allUserAnalysis['AboveAUAS']==1])
allUserAnalysis['AboveAUAS : 0'] = len(allUserAnalysis[allUserAnalysis['AboveAUAS']==0])
allUserAnalysis

/var/folders/gm/w1dhc0qj1x72sgdrxk_zc4jw0000gn/T/ipykernel_11487/3092088585.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  allUserAnalysis['AboveAUAS : 1'] = len(allUserAnalysis[allUserAnalysis['AboveAUAS']==1])
/var/folders/gm/w1dhc0qj1x72sgdrxk_zc4jw0000gn/T/ipykernel_11487/3092088585.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  allUserAnalysis['AboveAUAS : 0'] = len(allUserAnalysis[allUserAnalysis['AboveAUAS']==0])


series_id  Valid Night Count           AllUserAvgSleep  AboveAUAS  \
0    038441c925bb          17.806691 0 days 08:37:27.717387773          1   
1    03d92c9f6f8a          17.806691 0 days 08:37:27.717387773          1   
2    0402a003dae9          17.806691 0 days 08:37:27.717387773          0   
3    04f547b8017d          17.806691 0 days 08:37:27.717387773          1   
4    05e1944c3818          17.806691 0 days 08:37:27.717387773          0   
..            ...                ...                       ...        ...   
264  fa149c3c4bde          17.806691 0 days 08:37:27.717387773          1   
265  fb223ed2278c          17.806691 0 days 08:37:27.717387773          1   
266  fbf33b1a2c10          17.806691 0 days 08:37:27.717387773          0   
267  fcca183903b7          17.806691 0 days 08:37:27.717387773          0   
268  fe90110788d2          17.806691 0 days 08:37:27.717387773          1   

     AboveAUAS : 1  AboveAUAS : 0  
0              151            118  
1              151            118  
2              151            118  
3              151            118  
4              151            118  
..             ...            ...  
264            151            118  
265            151            118  
266            151            118  
267            151            118  
268            151            118  

[269 rows x 6 columns]

In [164]:
allUserAnalysis['UserCount']=len(allUserAnalysis['series_id'])
allUserAnalysis

/var/folders/gm/w1dhc0qj1x72sgdrxk_zc4jw0000gn/T/ipykernel_11487/1773977106.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  allUserAnalysis['UserCount']=len(allUserAnalysis['series_id'])


series_id  Valid Night Count           AllUserAvgSleep  AboveAUAS  \
0    038441c925bb          17.806691 0 days 08:37:27.717387773          1   
1    03d92c9f6f8a          17.806691 0 days 08:37:27.717387773          1   
2    0402a003dae9          17.806691 0 days 08:37:27.717387773          0   
3    04f547b8017d          17.806691 0 days 08:37:27.717387773          1   
4    05e1944c3818          17.806691 0 days 08:37:27.717387773          0   
..            ...                ...                       ...        ...   
264  fa149c3c4bde          17.806691 0 days 08:37:27.717387773          1   
265  fb223ed2278c          17.806691 0 days 08:37:27.717387773          1   
266  fbf33b1a2c10          17.806691 0 days 08:37:27.717387773          0   
267  fcca183903b7          17.806691 0 days 08:37:27.717387773          0   
268  fe90110788d2          17.806691 0 days 08:37:27.717387773          1   

     AboveAUAS : 1  AboveAUAS : 0  UserCount  
0              151            118        269  
1              151            118        269  
2              151            118        269  
3              151            118        269  
4              151            118        269  
..             ...            ...        ...  
264            151            118        269  
265            151            118        269  
266            151            118        269  
267            151            118        269  
268            151            118        269  

[269 rows x 7 columns]

In [170]:
allUserAnalysis=allUserAnalysis.iloc[[0]]
allUserAnalysis=allUserAnalysis.drop(['series_id','AboveAUAS'],axis=1)
allUserAnalysis

Valid Night Count           AllUserAvgSleep  AboveAUAS : 1  AboveAUAS : 0  \
0          17.806691 0 days 08:37:27.717387773            151            118   

   UserCount  
0        269

In [181]:
allUserAnalysis.to_csv('all_user_overall_summary.csv', index = False)

### 사람 추출하기

In [182]:
#1인 사람들
abovePeople=new_train_events.reset_index(drop=False,inplace=False)
abovePeople

AllUserAvgSleep     series_id  Valid Night Count  \
0    0 days 08:37:27.717387773  038441c925bb               19.0   
1    0 days 08:37:27.717387773  038441c925bb               19.0   
2    0 days 08:37:27.717387773  038441c925bb               19.0   
3    0 days 08:37:27.717387773  038441c925bb               19.0   
4    0 days 08:37:27.717387773  038441c925bb               19.0   
...                        ...           ...                ...   
9575 0 days 08:37:27.717387773  fe90110788d2               33.0   
9576 0 days 08:37:27.717387773  fe90110788d2               33.0   
9577 0 days 08:37:27.717387773  fe90110788d2               33.0   
9578 0 days 08:37:27.717387773  fe90110788d2               33.0   
9579 0 days 08:37:27.717387773  fe90110788d2               33.0   

                Avg sleep time  AboveAUAS      sleep time  night  event  \
0    0 days 08:46:41.052631578          1 0 days 08:15:00      1      1   
1    0 days 08:46:41.052631578          1 0 days 08:15:00      1      0   
2    0 days 08:46:41.052631578          1 0 days 10:04:00      2      1   
3    0 days 08:46:41.052631578          1 0 days 10:04:00      2      0   
4    0 days 08:46:41.052631578          1 0 days 06:07:00      3      1   
...                        ...        ...             ...    ...    ...   
9575 0 days 08:56:45.454545454          1 0 days 11:23:00     32      0   
9576 0 days 08:56:45.454545454          1 0 days 05:37:00     33      1   
9577 0 days 08:56:45.454545454          1 0 days 05:37:00     33      0   
9578 0 days 08:56:45.454545454          1 0 days 09:42:00     34      1   
9579 0 days 08:56:45.454545454          1 0 days 09:42:00     34      0   

          step                  timestamp   anglez    enmo  
0       4992.0  2018-08-14 22:26:00-04:00 -78.6906  0.0099  
1      10932.0  2018-08-15 06:41:00-04:00 -61.5782  0.0263  
2      20244.0  2018-08-15 19:37:00-04:00  -6.3874  0.0182  
3      27492.0  2018-08-16 05:41:00-04:00 -45.3551  0.0165  
4      39996.0  2018-08-16 23:03:00-04:00  -1.7867  0.0000  
...        ...                        ...      ...     ...  
9575  547152.0  2017-09-05 09:26:00-04:00 -56.7689  0.0042  
9576  556560.0  2017-09-05 22:30:00-04:00 -16.5867  0.0000  
9577  560604.0  2017-09-06 04:07:00-04:00 -43.0509  0.0000  
9578  574620.0  2017-09-06 23:35:00-04:00   1.5191  0.0066  
9579  581604.0  2017-09-07 09:17:00-04:00  79.0538  0.0000  

[9580 rows x 12 columns]

In [185]:
import numpy as np

In [188]:
abovePeople1=abovePeople[abovePeople['AboveAUAS'] == 1]
abovePeople1

AllUserAvgSleep     series_id  Valid Night Count  \
0    0 days 08:37:27.717387773  038441c925bb               19.0   
1    0 days 08:37:27.717387773  038441c925bb               19.0   
2    0 days 08:37:27.717387773  038441c925bb               19.0   
3    0 days 08:37:27.717387773  038441c925bb               19.0   
4    0 days 08:37:27.717387773  038441c925bb               19.0   
...                        ...           ...                ...   
9575 0 days 08:37:27.717387773  fe90110788d2               33.0   
9576 0 days 08:37:27.717387773  fe90110788d2               33.0   
9577 0 days 08:37:27.717387773  fe90110788d2               33.0   
9578 0 days 08:37:27.717387773  fe90110788d2               33.0   
9579 0 days 08:37:27.717387773  fe90110788d2               33.0   

                Avg sleep time  AboveAUAS      sleep time  night  event  \
0    0 days 08:46:41.052631578          1 0 days 08:15:00      1      1   
1    0 days 08:46:41.052631578          1 0 days 08:15:00      1      0   
2    0 days 08:46:41.052631578          1 0 days 10:04:00      2      1   
3    0 days 08:46:41.052631578          1 0 days 10:04:00      2      0   
4    0 days 08:46:41.052631578          1 0 days 06:07:00      3      1   
...                        ...        ...             ...    ...    ...   
9575 0 days 08:56:45.454545454          1 0 days 11:23:00     32      0   
9576 0 days 08:56:45.454545454          1 0 days 05:37:00     33      1   
9577 0 days 08:56:45.454545454          1 0 days 05:37:00     33      0   
9578 0 days 08:56:45.454545454          1 0 days 09:42:00     34      1   
9579 0 days 08:56:45.454545454          1 0 days 09:42:00     34      0   

          step                  timestamp   anglez    enmo  
0       4992.0  2018-08-14 22:26:00-04:00 -78.6906  0.0099  
1      10932.0  2018-08-15 06:41:00-04:00 -61.5782  0.0263  
2      20244.0  2018-08-15 19:37:00-04:00  -6.3874  0.0182  
3      27492.0  2018-08-16 05:41:00-04:00 -45.3551  0.0165  
4      39996.0  2018-08-16 23:03:00-04:00  -1.7867  0.0000  
...        ...                        ...      ...     ...  
9575  547152.0  2017-09-05 09:26:00-04:00 -56.7689  0.0042  
9576  556560.0  2017-09-05 22:30:00-04:00 -16.5867  0.0000  
9577  560604.0  2017-09-06 04:07:00-04:00 -43.0509  0.0000  
9578  574620.0  2017-09-06 23:35:00-04:00   1.5191  0.0066  
9579  581604.0  2017-09-07 09:17:00-04:00  79.0538  0.0000  

[5444 rows x 12 columns]

In [190]:
abovePeople1.drop('AllUserAvgSleep',axis=1,inplace=True)
abovePeople1

/var/folders/gm/w1dhc0qj1x72sgdrxk_zc4jw0000gn/T/ipykernel_11487/1901554705.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  abovePeople1.drop('AllUserAvgSleep',axis=1,inplace=True)


series_id  Valid Night Count            Avg sleep time  AboveAUAS  \
0     038441c925bb               19.0 0 days 08:46:41.052631578          1   
1     038441c925bb               19.0 0 days 08:46:41.052631578          1   
2     038441c925bb               19.0 0 days 08:46:41.052631578          1   
3     038441c925bb               19.0 0 days 08:46:41.052631578          1   
4     038441c925bb               19.0 0 days 08:46:41.052631578          1   
...            ...                ...                       ...        ...   
9575  fe90110788d2               33.0 0 days 08:56:45.454545454          1   
9576  fe90110788d2               33.0 0 days 08:56:45.454545454          1   
9577  fe90110788d2               33.0 0 days 08:56:45.454545454          1   
9578  fe90110788d2               33.0 0 days 08:56:45.454545454          1   
9579  fe90110788d2               33.0 0 days 08:56:45.454545454          1   

          sleep time  night  event      step                  timestamp  \
0    0 days 08:15:00      1      1    4992.0  2018-08-14 22:26:00-04:00   
1    0 days 08:15:00      1      0   10932.0  2018-08-15 06:41:00-04:00   
2    0 days 10:04:00      2      1   20244.0  2018-08-15 19:37:00-04:00   
3    0 days 10:04:00      2      0   27492.0  2018-08-16 05:41:00-04:00   
4    0 days 06:07:00      3      1   39996.0  2018-08-16 23:03:00-04:00   
...              ...    ...    ...       ...                        ...   
9575 0 days 11:23:00     32      0  547152.0  2017-09-05 09:26:00-04:00   
9576 0 days 05:37:00     33      1  556560.0  2017-09-05 22:30:00-04:00   
9577 0 days 05:37:00     33      0  560604.0  2017-09-06 04:07:00-04:00   
9578 0 days 09:42:00     34      1  574620.0  2017-09-06 23:35:00-04:00   
9579 0 days 09:42:00     34      0  581604.0  2017-09-07 09:17:00-04:00   

       anglez    enmo  
0    -78.6906  0.0099  
1    -61.5782  0.0263  
2     -6.3874  0.0182  
3    -45.3551  0.0165  
4     -1.7867  0.0000  
...       ...     ...  
9575 -56.7689  0.0042  
9576 -16.5867  0.0000  
9577 -43.0509  0.0000  
9578   1.5191  0.0066  
9579  79.0538  0.0000  

[5444 rows x 11 columns]

In [191]:
user_list=list(abovePeople1['series_id'].unique())
user_list

['038441c925bb',
 '03d92c9f6f8a',
 '04f547b8017d',
 '062cae666e2a',
 '062dbd4c95e6',
 '0a96f4993bd7',
 '0cfc06c129cc',
 '0ec9fc461819',
 '0f572d690310',
 '10469f6765bf',
 '10f8bc1f7b07',
 '12d01911d509',
 '1319a1935f48',
 '137771d19ca2',
 '13b4d6a01d27',
 '148471991ffb',
 '1716cd4163b2',
 '1762ab70ec76',
 '188d4b7cd28b',
 '18a0ca03431d',
 '1955d568d987',
 '1b92be89db4c',
 '1e6717d93c1d',
 '1f96b9668bdf',
 '207eded97727',
 '25e2b3dd9c3b',
 '280e08693c6d',
 '2b0a1fa8eba8',
 '2f7504d0f426',
 '2fbbee1a38e3',
 '31011ade7c0a',
 '3318a0e3ed6f',
 '33ceeba8918a',
 '361366da569e',
 '3665c86afaf5',
 '3a9a9dc2cbd9',
 '3be1545083b7',
 '3be2f86c3e45',
 '3c336d6ba566',
 '3df0da2e5966',
 '416354edd92a',
 '4a31811f3558',
 '4ac356361be9',
 '4b45c36f8f5a',
 '519ae2d858b0',
 '51fdcc8d9fe7',
 '559ffb7c166a',
 '55a47ff9dc8a',
 '55b7f5c99930',
 '599ca4ed791b',
 '5acc9d63b5fd',
 '5c088d7e916c',
 '5c55a5e717d6',
 '5f76965e10cf',
 '5ffd5e1e81ac',
 '601559e1777d',
 '612aa8ba44e2',
 '67f5fc60e494',
 '694faf956ebf

In [192]:
abovePeople1['series_id'][0]

'038441c925bb'

In [202]:
user_list[76]

'8becc76ea607'

In [197]:
abovePeople1['series_id'].unique

5444

In [196]:
abovePeople1['series_id'][0]

'038441c925bb'

In [200]:
aboveRandUser=list(np.random.randint(len(user_list)+1,size=5))
aboveRandUser

[76, 117, 82, 138, 130]

In [206]:
collected_1 = pd.DataFrame()

for n in aboveRandUser:
    rn1 = abovePeople1.loc[abovePeople1['series_id'] == user_list[n]]
    collected_1 = pd.concat([collected_1, rn1], ignore_index=True)
collected_1

series_id  Valid Night Count            Avg sleep time  AboveAUAS  \
0    8becc76ea607               13.0 0 days 09:42:55.384615384          1   
1    8becc76ea607               13.0 0 days 09:42:55.384615384          1   
2    8becc76ea607               13.0 0 days 09:42:55.384615384          1   
3    8becc76ea607               13.0 0 days 09:42:55.384615384          1   
4    8becc76ea607               13.0 0 days 09:42:55.384615384          1   
..            ...                ...                       ...        ...   
231  e0686434d029               32.0    0 days 08:55:05.625000          1   
232  e0686434d029               32.0    0 days 08:55:05.625000          1   
233  e0686434d029               32.0    0 days 08:55:05.625000          1   
234  e0686434d029               32.0    0 days 08:55:05.625000          1   
235  e0686434d029               32.0    0 days 08:55:05.625000          1   

         sleep time  night  event      step                  timestamp  \
0   0 days 09:43:00      1      1    7836.0  2019-01-14 21:53:00-05:00   
1   0 days 09:43:00      1      0   14832.0  2019-01-15 07:36:00-05:00   
2   0 days 09:44:00      2      1   25152.0  2019-01-15 21:56:00-05:00   
3   0 days 09:44:00      2      0   32160.0  2019-01-16 07:40:00-05:00   
4   0 days 09:15:00      3      1   42384.0  2019-01-16 21:52:00-05:00   
..              ...    ...    ...       ...                        ...   
231 0 days 08:48:00     31      0  528732.0  2018-05-07 06:36:00-04:00   
232 0 days 08:15:00     32      1  539952.0  2018-05-07 22:11:00-04:00   
233 0 days 08:15:00     32      0  545892.0  2018-05-08 06:26:00-04:00   
234 0 days 10:30:00     33      1  557496.0  2018-05-08 22:33:00-04:00   
235 0 days 10:30:00     33      0  565056.0  2018-05-09 09:03:00-04:00   

      anglez    enmo  
0   -58.4029  0.0330  
1   -83.9227  0.0393  
2   -49.6805  0.0201  
3   -52.6917  0.0296  
4    14.8078  0.0000  
..       ...     ...  
231  27.7320  0.0000  
232 -15.5248  0.0047  
233  83.8906  0.0017  
234 -14.7621  0.0000  
235  10.3908  0.0338  

[236 rows x 11 columns]

In [208]:
collected_1['series_id'].unique()

array(['8becc76ea607', 'd043c0ca71cd', '99b829cbad2d', 'e8d0a37c3eba',
       'e0686434d029'], dtype=object)

In [209]:
#1미만인 사람들
DownPeople=new_train_events.reset_index(drop=False,inplace=False)
DownPeople=DownPeople[DownPeople['AboveAUAS'] == 0]
DownPeople

AllUserAvgSleep     series_id  Valid Night Count  \
54   0 days 08:37:27.717387773  0402a003dae9               18.0   
55   0 days 08:37:27.717387773  0402a003dae9               18.0   
56   0 days 08:37:27.717387773  0402a003dae9               18.0   
57   0 days 08:37:27.717387773  0402a003dae9               18.0   
58   0 days 08:37:27.717387773  0402a003dae9               18.0   
...                        ...           ...                ...   
9509 0 days 08:37:27.717387773  fcca183903b7               35.0   
9510 0 days 08:37:27.717387773  fcca183903b7               35.0   
9511 0 days 08:37:27.717387773  fcca183903b7               35.0   
9512 0 days 08:37:27.717387773  fcca183903b7               35.0   
9513 0 days 08:37:27.717387773  fcca183903b7               35.0   

                Avg sleep time  AboveAUAS      sleep time  night  event  \
54   0 days 06:33:46.666666666          0 0 days 06:22:00      1      1   
55   0 days 06:33:46.666666666          0 0 days 06:22:00      1      0   
56   0 days 06:33:46.666666666          0 0 days 07:44:00      2      1   
57   0 days 06:33:46.666666666          0 0 days 07:44:00      2      0   
58   0 days 06:33:46.666666666          0 0 days 02:59:00      6      1   
...                        ...        ...             ...    ...    ...   
9509 0 days 08:30:22.285714285          0 0 days 08:24:00     33      0   
9510 0 days 08:30:22.285714285          0 0 days 09:19:00     34      1   
9511 0 days 08:30:22.285714285          0 0 days 09:19:00     34      0   
9512 0 days 08:30:22.285714285          0 0 days 09:26:00     35      1   
9513 0 days 08:30:22.285714285          0 0 days 09:26:00     35      0   

          step                  timestamp   anglez    enmo  
54      8364.0  2018-12-19 00:22:00-05:00  29.0663  0.0000  
55     12948.0  2018-12-19 06:44:00-05:00  42.9404  0.0008  
56     24396.0  2018-12-19 22:38:00-05:00   1.8807  0.0099  
57     29964.0  2018-12-20 06:22:00-05:00 -39.7638  0.0507  
58     94032.0  2018-12-23 23:21:00-05:00 -77.3549  0.0303  
...        ...                        ...      ...     ...  
9509  565824.0  2019-04-28 06:52:00-04:00  -7.3872  0.0033  
9510  577344.0  2019-04-28 22:52:00-04:00 -44.7628  0.0000  
9511  584052.0  2019-04-29 08:11:00-04:00 -68.2084  0.0341  
9512  595344.0  2019-04-29 23:52:00-04:00 -12.1260  0.0517  
9513  602136.0  2019-04-30 09:18:00-04:00  82.3611  0.0000  

[4136 rows x 12 columns]

In [210]:
DownPeople.drop('AllUserAvgSleep',axis=1,inplace=True)

In [211]:
DownList=list(DownPeople['series_id'].unique())
DownList

['0402a003dae9',
 '05e1944c3818',
 '08db4255286f',
 '0cd1e3d0ed95',
 '0ce74d6d2106',
 '0d0ad1e77851',
 '0dee4fda51c3',
 '0ef7d94fde99',
 '1087d7b0ff2e',
 '137b99e936ab',
 '154fe824ed87',
 '16fe2798ed0f',
 '18b61dd5aae8',
 '1c7c0bad1263',
 '1d4569cbac0f',
 '2654a87be968',
 '27f09a6a858f',
 '292a75c0b94e',
 '29c75c018220',
 '29d3469bd15d',
 '2b8d87addea9',
 '2cd2340ca14d',
 '2e9ced2c7976',
 '3452b878e596',
 '349c5562ee2c',
 '35826366dfc7',
 '3664fe9233f9',
 '3aceb17ef7bd',
 '3d53bfea61d6',
 '405df1b41f9f',
 '40dce6018935',
 '449766346eb1',
 '44a41bba1ee7',
 '44d8c02b369e',
 '4743bdde25df',
 '483d6545417f',
 '4ab54be1a403',
 '4feda0596965',
 '51b23d177971',
 '51c49c540b4e',
 '5aad18e7ce64',
 '5e816f11f5c3',
 '5f40907ec171',
 '5f94bb3e1bed',
 '60d31b0bec3b',
 '60e51cad2ffb',
 '653622ac8363',
 '655f19eabf1e',
 '6a4cd123bd69',
 '6ca4f4fca6a2',
 '72d2234e84e4',
 '7476c0bd18d2',
 '752900afe3a6',
 '76237b9406d5',
 '77ca4db83644',
 '7822ee8fe3ec',
 '785c9ca4eff7',
 '7df249527c63',
 '7fd4284b7ee8

In [212]:
len(DownList)

118

In [217]:
downRandUser=list(np.random.randint(len(DownList)+1,size=5))
downRandUser

[52, 53, 30, 74, 6]

In [215]:
DownPeople

series_id  Valid Night Count            Avg sleep time  AboveAUAS  \
54    0402a003dae9               18.0 0 days 06:33:46.666666666          0   
55    0402a003dae9               18.0 0 days 06:33:46.666666666          0   
56    0402a003dae9               18.0 0 days 06:33:46.666666666          0   
57    0402a003dae9               18.0 0 days 06:33:46.666666666          0   
58    0402a003dae9               18.0 0 days 06:33:46.666666666          0   
...            ...                ...                       ...        ...   
9509  fcca183903b7               35.0 0 days 08:30:22.285714285          0   
9510  fcca183903b7               35.0 0 days 08:30:22.285714285          0   
9511  fcca183903b7               35.0 0 days 08:30:22.285714285          0   
9512  fcca183903b7               35.0 0 days 08:30:22.285714285          0   
9513  fcca183903b7               35.0 0 days 08:30:22.285714285          0   

          sleep time  night  event      step                  timestamp  \
54   0 days 06:22:00      1      1    8364.0  2018-12-19 00:22:00-05:00   
55   0 days 06:22:00      1      0   12948.0  2018-12-19 06:44:00-05:00   
56   0 days 07:44:00      2      1   24396.0  2018-12-19 22:38:00-05:00   
57   0 days 07:44:00      2      0   29964.0  2018-12-20 06:22:00-05:00   
58   0 days 02:59:00      6      1   94032.0  2018-12-23 23:21:00-05:00   
...              ...    ...    ...       ...                        ...   
9509 0 days 08:24:00     33      0  565824.0  2019-04-28 06:52:00-04:00   
9510 0 days 09:19:00     34      1  577344.0  2019-04-28 22:52:00-04:00   
9511 0 days 09:19:00     34      0  584052.0  2019-04-29 08:11:00-04:00   
9512 0 days 09:26:00     35      1  595344.0  2019-04-29 23:52:00-04:00   
9513 0 days 09:26:00     35      0  602136.0  2019-04-30 09:18:00-04:00   

       anglez    enmo  
54    29.0663  0.0000  
55    42.9404  0.0008  
56     1.8807  0.0099  
57   -39.7638  0.0507  
58   -77.3549  0.0303  
...       ...     ...  
9509  -7.3872  0.0033  
9510 -44.7628  0.0000  
9511 -68.2084  0.0341  
9512 -12.1260  0.0517  
9513  82.3611  0.0000  

[4136 rows x 11 columns]

In [218]:
DownList[52]

'752900afe3a6'

In [219]:
collected_0 = pd.DataFrame()

for n in downRandUser:
    rn0 = DownPeople.loc[DownPeople['series_id'] == DownList[n],:]
    collected_0 = pd.concat([collected_0, rn0], ignore_index=True)
collected_0

series_id  Valid Night Count            Avg sleep time  AboveAUAS  \
0    752900afe3a6               21.0 0 days 08:15:48.571428571          0   
1    752900afe3a6               21.0 0 days 08:15:48.571428571          0   
2    752900afe3a6               21.0 0 days 08:15:48.571428571          0   
3    752900afe3a6               21.0 0 days 08:15:48.571428571          0   
4    752900afe3a6               21.0 0 days 08:15:48.571428571          0   
..            ...                ...                       ...        ...   
191  0dee4fda51c3               20.0           0 days 06:20:15          0   
192  0dee4fda51c3               20.0           0 days 06:20:15          0   
193  0dee4fda51c3               20.0           0 days 06:20:15          0   
194  0dee4fda51c3               20.0           0 days 06:20:15          0   
195  0dee4fda51c3               20.0           0 days 06:20:15          0   

         sleep time  night  event      step                  timestamp  \
0   0 days 09:06:00      1      1    3420.0  2017-11-21 21:15:00-05:00   
1   0 days 09:06:00      1      0    9972.0  2017-11-22 06:21:00-05:00   
2   0 days 08:08:00      2      1   21372.0  2017-11-22 22:11:00-05:00   
3   0 days 08:08:00      2      0   27228.0  2017-11-23 06:19:00-05:00   
4   0 days 08:01:00      3      1   38532.0  2017-11-23 22:01:00-05:00   
..              ...    ...    ...       ...                        ...   
191 0 days 05:56:00     21      0  356976.0  2019-01-25 04:48:00-05:00   
192 0 days 05:00:00     23      1  390960.0  2019-01-27 04:00:00-05:00   
193 0 days 05:00:00     23      0  394560.0  2019-01-27 09:00:00-05:00   
194 0 days 04:22:00     24      1  407880.0  2019-01-28 03:30:00-05:00   
195 0 days 04:22:00     24      0  411024.0  2019-01-28 07:52:00-05:00   

      anglez    enmo  
0    41.0932  0.0029  
1   -50.3950  0.3559  
2     3.4592  0.0108  
3    11.2048  0.0000  
4   -22.7348  0.0315  
..       ...     ...  
191  17.5250  0.0109  
192  -8.6461  0.0112  
193 -52.7052  0.0100  
194  11.8506  0.0000  
195  24.5208  0.0074  

[196 rows x 11 columns]

In [220]:
collected_0['series_id'].unique()

array(['752900afe3a6', '76237b9406d5', '40dce6018935', 'a261bc4b7470',
       '0dee4fda51c3'], dtype=object)

In [221]:
someUser=pd.concat([collected_1, collected_0], axis = 0) 
someUser

series_id  Valid Night Count            Avg sleep time  AboveAUAS  \
0    8becc76ea607               13.0 0 days 09:42:55.384615384          1   
1    8becc76ea607               13.0 0 days 09:42:55.384615384          1   
2    8becc76ea607               13.0 0 days 09:42:55.384615384          1   
3    8becc76ea607               13.0 0 days 09:42:55.384615384          1   
4    8becc76ea607               13.0 0 days 09:42:55.384615384          1   
..            ...                ...                       ...        ...   
191  0dee4fda51c3               20.0           0 days 06:20:15          0   
192  0dee4fda51c3               20.0           0 days 06:20:15          0   
193  0dee4fda51c3               20.0           0 days 06:20:15          0   
194  0dee4fda51c3               20.0           0 days 06:20:15          0   
195  0dee4fda51c3               20.0           0 days 06:20:15          0   

         sleep time  night  event      step                  timestamp  \
0   0 days 09:43:00      1      1    7836.0  2019-01-14 21:53:00-05:00   
1   0 days 09:43:00      1      0   14832.0  2019-01-15 07:36:00-05:00   
2   0 days 09:44:00      2      1   25152.0  2019-01-15 21:56:00-05:00   
3   0 days 09:44:00      2      0   32160.0  2019-01-16 07:40:00-05:00   
4   0 days 09:15:00      3      1   42384.0  2019-01-16 21:52:00-05:00   
..              ...    ...    ...       ...                        ...   
191 0 days 05:56:00     21      0  356976.0  2019-01-25 04:48:00-05:00   
192 0 days 05:00:00     23      1  390960.0  2019-01-27 04:00:00-05:00   
193 0 days 05:00:00     23      0  394560.0  2019-01-27 09:00:00-05:00   
194 0 days 04:22:00     24      1  407880.0  2019-01-28 03:30:00-05:00   
195 0 days 04:22:00     24      0  411024.0  2019-01-28 07:52:00-05:00   

      anglez    enmo  
0   -58.4029  0.0330  
1   -83.9227  0.0393  
2   -49.6805  0.0201  
3   -52.6917  0.0296  
4    14.8078  0.0000  
..       ...     ...  
191  17.5250  0.0109  
192  -8.6461  0.0112  
193 -52.7052  0.0100  
194  11.8506  0.0000  
195  24.5208  0.0074  

[432 rows x 11 columns]

In [224]:
someUser.to_csv('random_people_by_AboveAUAS.csv', index = False)